---
title: "CT26_DiffAnalysis"
author: "Mengshu"
date: "12/6/2018"
output: html_document
---

```{r}
library(Seurat)
library(SAVER)
library(dplyr)
library(ggplot2)
library(magrittr)
#Mac
source("/Volumes/Enterprise/FLX/Files_from_Gene/R_functions.r")
#Windows
#source("D:/FLX/Files_from_Gene/R_functions.r")
```

Load data into table
```{r}

F1 <- Read10X_h5("F1_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE) # 288 cells, 2 min 
F2 <- Read10X_h5("F2_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE) # 273 cells
F3 <- Read10X_h5("F3_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE) # 252 cells

D1 <- Read10X_h5("D1_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE) # 188 cells, 2 min 
D2 <- Read10X_h5("D2_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE) # 287 cells
D3 <- Read10X_h5("D3_filtered_gene_bc_matrices_h5.h5", ensg.names = FALSE)
dim(F1)
```

Merge F Replicates together
```{r}
Fresh <- CreateSeuratObject(F1,normalization.method = NULL, do.scale = FALSE, do.center = FALSE,
  names.field = 1,display.progress = TRUE, project="F1")
Fresh <- AddSamples(object = Fresh, new.data = F2, add.cell.id = "F2")
Fresh <- AddSamples(object = Fresh, new.data = F3, add.cell.id = "F3")
Fresh@meta.data$stim <- "fresh"
Fresh <- NormalizeData(Fresh)
Fresh <- ScaleData(Fresh, display.progress = FALSE)
#Check the three replicates
table(Fresh@meta.data$orig.ident)
```

Merge D Replicates together
```{r}
D <- CreateSeuratObject(D1, normalization.method = NULL, do.scale = FALSE, do.center = FALSE,
  names.field = 1,display.progress = TRUE, project="D1")
D <- AddSamples(object = D, new.data = D2, add.cell.id = "D2")
D@meta.data$stim <- "dmso"
D <- NormalizeData(D)
D <- ScaleData(D, display.progress = FALSE)
#Check the three replicates
table(D@meta.data$orig.ident)
```
####
Compare Good DMSO replicate to the bad one 1 vs 3
######

```{r}
good <- CreateSeuratObject(D1, normalization.method = NULL, do.scale = FALSE, do.center = FALSE, names.field = 1,display.progress = TRUE, project="D1")
good@meta.data$stim <- "good"
good <- NormalizeData(good)
good <- ScaleData(good, display.progress = FALSE)

bad <- CreateSeuratObject(D3, normalization.method = NULL, do.scale = FALSE, do.center = FALSE, names.field = 1,display.progress = TRUE, project="D3")
bad@meta.data$stim <- "bad"
bad <- NormalizeData(bad)
bad <- ScaleData(bad, display.progress = FALSE)
```

Gene selection for CCA input
```{r}
# Gene selection for input to CCA
Fresh <- FindVariableGenes(Fresh, do.plot = FALSE)
D <- FindVariableGenes(D, do.plot = FALSE)
g.1 <- head(rownames(Fresh@hvg.info), 1000)
g.2 <- head(rownames(D@hvg.info), 1000)
genes.use <- unique(c(g.1, g.2))
genes.use <- intersect(genes.use, rownames(Fresh@scale.data))
genes.use <- intersect(genes.use, rownames(D@scale.data))
str(genes.use)

good <- FindVariableGenes(good, do.plot = FALSE)
bad <- FindVariableGenes(bad, do.plot = FALSE)
good.1 <- head(rownames(good@hvg.info), 1000)
bad.2 <- head(rownames(bad@hvg.info), 1000)
genes.use2 <- unique(c(good.1, bad.2))
genes.use2 <- intersect(genes.use2, rownames(good@scale.data))
genes.use2 <- intersect(genes.use2, rownames(bad@scale.data))
str(genes.use2)

```

################
Perform Canonical Correlation Analysis CCA
Find common sources of variation
this also combines the two objects into one
################

```{r}
CT26.combined <- RunCCA(Fresh, D, genes.use = genes.use, num.cc = 30)

# visualize results of CCA plot CC1 versus CC2 and look at a violin plot
p1 <- DimPlot(object = CT26.combined, reduction.use = "cca", group.by = "stim", 
    pt.size = 0.5, do.return = TRUE)
p2 <- VlnPlot(object = CT26.combined, features.plot = "CC1", group.by = "stim", 
    do.return = TRUE)
plot_grid(p1, p2)


D.combined <- RunCCA(good, bad, genes.use = genes.use2, num.cc = 30)
p.D1 <- DimPlot(object = D.combined, reduction.use = "cca", group.by = "stim", 
    pt.size = 0.5, do.return = TRUE)
p.D2 <- VlnPlot(object = D.combined, features.plot = "CC1", group.by = "stim", 
    do.return = TRUE)
plot_grid(p.D1, p.D2) 
ggsave(file="D1vsD3_CCA+results.jpg", width=8, height=4, units="in", plot=last_plot(), dpi=600)
```
Lymphoid markers in CC1
Myeloid markers in CC2
There is a shift, so the two treatments are causing differences in expression
CC1 is differentiated by Myeloid markers on the high score end, Lymphoid markers on the low score end
CC2 is RP proteins on the high score end, mitochondrial apoptotic markers on the low score end

```{r fig.height=10, fig.width=10}
PrintDim(object = CT26.combined, reduction.type = "cca", dims.print = 1:2, 
    genes.print = 10)

DimHeatmap(object = CT26.combined, reduction.type = "cca", cells.use = 500, 
    dim.use = 1:9, do.balanced = TRUE)

PrintDim(object = D.combined, reduction.type = "cca", dims.print = 1:2, 
    genes.print = 10)

DimHeatmap(object = D.combined, reduction.type = "cca", cells.use = 500, 
    dim.use = 1:9, do.balanced = TRUE)
```
######################
#######################
ALign the CCA subspaces
Aligns the different treatment groups so that they cluster closer together by cell identity rather than global treatment differeneces
Visualize the aligned CCA after alignment
```{r}
CT26.combined <- AlignSubspace(CT26.combined, reduction.type = "cca", grouping.var = "stim", 
    dims.align = 1:9)


D.combined <- AlignSubspace(D.combined, reduction.type = "cca", grouping.var = "stim", 
    dims.align = 1:9)

p1 <- VlnPlot(object = D.combined, features.plot = "ACC1", group.by = "stim", 
    do.return = TRUE)
p2 <- VlnPlot(object = D.combined, features.plot = "ACC2", group.by = "stim", 
    do.return = TRUE)
plot_grid(p1, p2)
```

We can visualize the aligned CCA and perform an integrated analysis.

```{r}
p1 <- VlnPlot(object = CT26.combined, features.plot = "ACC1", group.by = "stim", 
    do.return = TRUE)
p2 <- VlnPlot(object = CT26.combined, features.plot = "ACC2", group.by = "stim", 
    do.return = TRUE)
plot_grid(p1, p2)

p3 <- VlnPlot(object = D.combined, features.plot = "ACC1", group.by = "stim", 
    do.return = TRUE)
p4 <- VlnPlot(object = D.combined, features.plot = "ACC2", group.by = "stim", 
    do.return = TRUE)
plot_grid(p3, p4)
```
```{r}
# t-SNE and Clustering
CT26.combined <- RunTSNE(CT26.combined, reduction.use = "cca.aligned", dims.use = 1:9, 
    do.fast = T)
CT26.combined <- FindClusters(CT26.combined, reduction.type = "cca.aligned", 
    resolution = 0.6, dims.use = 1:9)
# Visualization
p1 <- TSNEPlot(CT26.combined, do.return = T, pt.size = 0.5, group.by = "stim")
p2 <- TSNEPlot(CT26.combined, do.label = T, do.return = T, pt.size = 0.5)
plot_grid(p1, p2) +
ggsave("DMSOvsFresh_TSNE.jpg", width= 10, height=6, dpi=600, plot= last_plot(), units = "in")


# t-SNE and Clustering
D.combined <- RunTSNE(D.combined, reduction.use = "cca.aligned", dims.use = 1:9, 
    do.fast = T)
D.combined <- FindClusters(D.combined, reduction.type = "cca.aligned", 
    resolution = 0.6, dims.use = 1:9)
# Visualization
p1.2 <- TSNEPlot(D.combined, do.return = T, pt.size = 0.5, group.by = "stim")
p2.2 <- TSNEPlot(D.combined, do.label = T, do.return = T, pt.size = 0.5)
plot_grid(p1.2, p2.2) +
ggsave("D1_vs_D3_TSNE.jpg", width= 9, height=4, dpi=600, plot= last_plot(), units = "in")
```


```{r fig.height=10, fig.width=16}
markers1 <- FindConservedMarkers(CT26.combined, ident.1 = 4, grouping.var = "stim", 
    print.bar = FALSE)
head(markers1)

markersD <- FindConservedMarkers(D.combined, ident.1 = 2, grouping.var = "stim", 
    print.bar = FALSE)
markersD %<>% arrange(-good_avg_logFC) 
head(markersD)

```


```{r fig.height=10, fig.width=15}
FeaturePlot(object = D.combined, features.plot = c("Cd3d","Cd8a","Cd4","Gzma","Cd14","Lyz2","Ccl7","Nkg7","Foxp3","Mki67","mt-Co3","Xist","Sox2","Col1a2","Cd79a"), min.cutoff = "q9", cols.use = c("lightgrey", 
    "blue"), pt.size = 0.5, no.legend=FALSE) +
ggsave("D1_D3_features_plot.jpg", width=12, height=7, units="in", dpi= 600, plot = last_plot())

```

Define plotting functions
```{r}
LabelPoint <- function(plot, genes, exp.mat, adj.x.t = 0, adj.y.t = 0, adj.x.s = 0, 
    adj.y.s = 0, text.size = 2.5, segment.size = 0.1) {
    for (i in genes) {
        x1 <- exp.mat[i, 1]
        y1 <- exp.mat[i, 2]
        plot <- plot + annotate("text", x = x1 + adj.x.t, y = y1 + adj.y.t, 
            label = i, size = text.size)
        plot <- plot + annotate("segment", x = x1 + adj.x.s, xend = x1, y = y1 + 
            adj.y.s, yend = y1, size = segment.size)
    }
    return(plot)
}

LabelUR <- function(plot, genes, exp.mat, adj.u.t = 0.1, adj.r.t = 0.15, adj.u.s = 0.05, 
    adj.r.s = 0.05, ...) {
    return(LabelPoint(plot, genes, exp.mat, adj.y.t = adj.u.t, adj.x.t = adj.r.t, 
        adj.y.s = adj.u.s, adj.x.s = adj.r.s, ...))
}

LabelUL <- function(plot, genes, exp.mat, adj.u.t = 0.1, adj.l.t = 0.15, adj.u.s = 0.05, 
    adj.l.s = 0.05, ...) {
    return(LabelPoint(plot, genes, exp.mat, adj.y.t = adj.u.t, adj.x.t = -adj.l.t, 
        adj.y.s = adj.u.s, adj.x.s = -adj.l.s, ...))
}
```


```{r}
t.cells <- SubsetData(CT26.combined, ident.use = 4, subset.raw = T)
t.cells <- SetAllIdent(t.cells, id = "stim")
avg.t.cells <- log1p(AverageExpression(t.cells, show.progress = FALSE))
avg.t.cells$gene <- rownames(avg.t.cells)

NK <- SubsetData(CT26.combined, ident.use = 1, subset.raw = T)
NK <- SetAllIdent(NK, id = "stim")
avg.NK <- log1p(AverageExpression(NK, show.progress = FALSE))
avg.NK$gene <- rownames(avg.NK)

Mono <- SubsetData(CT26.combined, ident.use = 2, subset.raw = T)
Mono <- SetAllIdent(Mono, id = "stim")
avg.Mono <- log1p(AverageExpression(Mono, show.progress = FALSE))
avg.Mono$gene <- rownames(avg.Mono)

DC <- SubsetData(CT26.combined, ident.use = 0, subset.raw = T)
DC <- SetAllIdent(DC, id = "stim")
avg.DC <- log1p(AverageExpression(DC, show.progress = FALSE))
avg.DC$gene <- rownames(avg.DC)

CT26 <- SubsetData(CT26.combined, ident.use = 3, subset.raw = T)
CT26 <- SetAllIdent(CT26, id = "stim")
avg.CT26 <- log1p(AverageExpression(CT26, show.progress = FALSE))
avg.CT26$gene <- rownames(avg.CT26)

B <- SubsetData(CT26.combined, ident.use = 5, subset.raw = T)
B <- SetAllIdent(B, id = "stim")
avg.B <- log1p(AverageExpression(B, show.progress = FALSE))
avg.B$gene <- rownames(avg.B)

p1 <- ggplot(avg.t.cells, aes(fresh, dmso)) + geom_point() + ggtitle("T Cells")
p2 <- ggplot(avg.NK, aes(fresh, dmso)) + geom_point() + ggtitle("NK")
p3 <- ggplot(avg.Mono, aes(fresh, dmso)) + geom_point() + ggtitle("Monocyte")
p4 <- ggplot(avg.DC, aes(fresh, dmso)) + geom_point() + ggtitle("DC")
p5 <- ggplot(avg.CT26, aes(fresh, dmso)) + geom_point() + ggtitle("CT26")
p6 <- ggplot(avg.B, aes(fresh, dmso)) + geom_point() + ggtitle("B")

avg.t.cells %>% mutate(ratio=dmso/fresh) %>% dfilter(dmso >= 0.8, fresh >= 0.8) %>% arrange((ratio)) %>% head
plot_grid(p1, p2, p3, p4, p5, p6)

ggsave("DMSOvsFresh_Scatter.jpg", width= 15, height=10, dpi=600, plot= last_plot(), units = "in")
```

```{r}
t.cells <- SubsetData(D.combined, ident.use = c("3_good", "3_bad"), subset.raw = T)
t.cells <- SetAllIdent(t.cells, id = "stim")
avg.t.cells <- log1p(AverageExpression(t.cells, show.progress = FALSE))
avg.t.cells$gene <- rownames(avg.t.cells)

NK <- SubsetData(D.combined, ident.use = c("4_good", "4_bad"), subset.raw = T)
NK <- SetAllIdent(NK, id = "stim")
avg.NK <- log1p(AverageExpression(NK, show.progress = FALSE))
avg.NK$gene <- rownames(avg.NK)

Mono <- SubsetData(D.combined, ident.use = c("1_good", "1_bad"), subset.raw = T)
Mono <- SetAllIdent(Mono, id = "stim")
avg.Mono <- log1p(AverageExpression(Mono, show.progress = FALSE))
avg.Mono$gene <- rownames(avg.Mono)

DC <- SubsetData(D.combined, ident.use = c("0_good", "0_bad"), subset.raw = T)
DC <- SetAllIdent(DC, id = "stim")
avg.DC <- log1p(AverageExpression(DC, show.progress = FALSE))
avg.DC$gene <- rownames(avg.DC)

CT26 <- SubsetData(D.combined, ident.use = c("5_good", "5_bad"), subset.raw = T)
CT26 <- SetAllIdent(CT26, id = "stim")
avg.CT26 <- log1p(AverageExpression(CT26, show.progress = FALSE))
avg.CT26$gene <- rownames(avg.CT26)

B <- SubsetData(D.combined, ident.use = c("2_good", "2_bad"), subset.raw = T)
B <- SetAllIdent(B, id = "stim")
avg.B <- log1p(AverageExpression(B, show.progress = FALSE))
avg.B$gene <- rownames(avg.B)

p1 <- ggplot(avg.t.cells, aes(good, bad)) + geom_point() + ggtitle("T Cells")
p2 <- ggplot(avg.NK, aes(good, bad)) + geom_point() + ggtitle("NK")
p3 <- ggplot(avg.Mono, aes(good, bad)) + geom_point() + ggtitle("Monocyte")
 
ggplot(avg.DC, aes(good, bad), do.hover= TRUE) + geom_point() + ggtitle("DC")
p5 <- ggplot(avg.CT26, aes(good, bad)) + geom_point() + ggtitle("CT26")
p6 <- ggplot(avg.B, aes(good, bad)) + geom_point() + ggtitle("B")

avg.t.cells %>% mutate(ratio=dmso/fresh) %>% dfilter(dmso >= 0.8, fresh >= 0.8) %>% arrange((ratio)) %>% head
plot_grid(p1, p2, p3, p4, p5, p6) +
ggsave("D1vsD3_Scatter.jpg", width= 15, height=10, dpi=600, plot= last_plot(), units = "in")
```

#####################
####################
Differential expression between experimental conditions
####################
##################
Cluster0: ITGAL is up in Good, RP proteins are different
Cluster1: tyk2 tyrosine kinase 2, DC markers Igsf6, setd1b Set1 homolog is up
Cluster2: Lcp1 lyphocyte cytoplasmic protein is down in good 
```{r}
D.combined@meta.data$celltype.stim <- paste0(D.combined@ident, "_", 
    D.combined@meta.data$stim) # name the groups 1_good, 1_bad etc. 
D.combined <- StashIdent(D.combined, save.name = "celltype")
D.combined <- SetAllIdent(D.combined, id = "celltype.stim")
what.happened <- FindMarkers(D.combined, ident.1 = "2_good", ident.2 = "2_bad", 
    print.bar = FALSE)
head(what.happened, 30)
```
```{r fig.height=20, fig.width=10}
FeatureHeatmap(D.combined, features.plot = c("Itgal", "Tyk2", "Lcp1", "Rps10","Mki67","mt-Co3", 
    ""), group.by = "stim", pt.size = 0.25, key.position = "top", 
    max.exp = 3)
```
```{r}
saveRDS(D.combined, file = "D1vsD3_working.rds")
D.combined <- readRDS("D1vsD3_working.rds")
```

