---
title: "Garvan Human Immunome"
author: "Mengshu"
date: "1/31/2019"
output: html_document
---

```{r}
library(dplyr)
library(magrittr)
library(ggplot2)
library(EDASeq)
source("/Volumes/Enterprise/FLX/Files_from_Gene/R_functions.r")
```


Download microarray dataset from Garvan Institute, the human immune transcriptome
HG-U133A] Affymetrix Human Genome U133A Array
Data is in SOFT format
Translate the spot names to gene names
Load SOFT datafile from NCBI GEO

```{r}

BiocManager::install("Biobase")
BiocManager::install("GEOquery")
library(GEOquery)
library(Biobase)
gse <- getGEO("GSE3982", destdir=".",GSEMatrix=FALSE)
#garvan <- getGEO("GSE3982", destdir=".",GSEMatrix=TRUE)
show(gse)
Meta(gse)
names(GSMList(gse))
GSMList(gse)[[1]]
```
Look at metadata
```{r}
names(GPLList(gse))
#eset <- GDS2eSet(gse,do.log2=TRUE)
gsmplatforms <- lapply(GSMList(gse),function(x) {Meta(x)$platform_id})
length(gsmplatforms)
gsmplatforms
gsmlist = Filter(function(gsm) {Meta(gsm)$platform_id=='GPL96'},GSMList(gse))
length(gsmlist)
gsmlist
Table(gsmlist[[32]])[1:5,]
Columns(gsmlist[[32]])[1:5,]
```

```{r}
# get the probeset ordering
probesets <- Table(GPLList(gse)[[1]])$ID
# make the data matrix from the VALUE columns from each GSM
# being careful to match the order of the probesets in the platform
# with those in the GSMs
data.matrix <- do.call('cbind',lapply(gsmlist,function(x) 
                                      {tab <- Table(x)
                                      return(tab$VALUE)  
                                      #mymatch <- match(probesets,tab$ID_REF)
                                       #return(tab$VALUE[mymatch])
                                     }))
data.matrix <- apply(data.matrix,2,function(x) {as.numeric(as.character(x))})
data.matrix <- log2(data.matrix)
data.matrix[1:5,]
dim(data.matrix)
```
I have no idea what this code is doing
```{r}
require(Biobase)
# go through the necessary steps to make a compliant ExpressionSet
rownames(data.matrix) <- probesets
my_colnames <- c(names(gsmlist[1:19]),names(gsmlist[21:32])) #somehow we lost a sample in the do.call cbind step, but it's a basophil sample so I just took its colname out
colnames(data.matrix) <- my_colnames
pdata <- data.frame(samples=my_colnames)
rownames(pdata) <- my_colnames
pheno <- as(pdata,"AnnotatedDataFrame")
eset2 <- new('ExpressionSet',exprs=data.matrix,phenoData=pheno)
eset2
```

```{r}
immu <- newSeqExpressionSet(data.matrix)

boxplot(immu, col="blue")
ggsave("Garvan_raw_signal.jpg", dpi=600, width=8, height=5, units="in", plot= last_plot())

immuNorm <- betweenLaneNormalization(immu, which="full")
garvan_norm <- normCounts(immuNorm)
rownames(garvan_norm) <- probesets
colnames(garvan_norm) <- my_colnames
boxplot(immuNorm, col="blue")
ggsave("Garvan_Norm_signal.jpg", dpi=600, width=8, height=5, units="in", plot= last_plot())
```

Look up the gene names

Problem is the tool returns
length(HUGO[,1]) 22283
length(OUT)    22300
length(unique(OUT$SYMBOL)) 
```{r}
source("http://bioconductor.org/biocLite.R")
biocLite("hgu95av2.db")
BiocManager::install("hgu133a2.db")
library("AnnotationDbi")
library("hgu95av2.db")    ##for Human
library("hgu133a2.db") 
PROBES<- as.character(row.names(garvan_norm))
OUT <- select(hgu133a2.db,keys= PROBES, columns=c("SYMBOL","GENENAME"),multiVals="first")
OUT[300:310,]
HUGO[300:310,]
HUGO <- OUT[!duplicated(OUT[,1]),]
length(HUGO[,1])
rownames(garvan_norm)[1:15]
rownames(garvan_norm) <- HUGO[,2]
```

Save file
```{r}
write.table(data.matrix, file="Garvan_Human_Immu.txt", sep="\t", row.names=TRUE, col.names=TRUE)
write.table(garvan_norm, file="Garvan_Human_Immu_Normalized.txt", sep="\t", row.names=TRUE, col.names=TRUE)
write.table(garvan_unique, file="Garvan_Human_Immu_NormUnique.txt", sep="\t", row.names=TRUE, col.names=TRUE)
write.table(HUGO, file="HG_U133Av2_HUGO.txt", sep="\t", col.names=TRUE)

garvan <- read.csv("Garvan_Human_Immu_Normalized.txt", header=TRUE,sep="\t", row.names=NULL)

garvan
```

There are duplicate rows for genes that have more than one probe, so take the average expression
garvan has 22283 lines
garvan_unique has 12996 lines
```{r}
garvan_unique <- garvan %>% group_by(row.names) %>% summarise_all(funs(mean))
garvan_unique %<>% dfilter(row.names != "NA")
my_genes <- garvan_unique[,1]
garvan_unique %<>% dselect(-row.names)
colnames(garvan_unique) <-  c("Eosinophils_control_rep1",	"Eosinophils_PMA 2h_rep1",	"Cord blood-derived mast cells_control_rep1",	"Cord blood-derived mast cells_IgE 2h_rep1",	"Dendritic cells_LPS 48h_rep1",	"Cord blood-derived mast cells_IgE 2h_rep2",	"Immature dendritic cells_rep1",	"Immature dendritic cells_rep2",	"Dendritic cells_LPS 48h_rep2",	"Macrophages_unstimulated_rep1",	"Macrophages_unstimulated_rep2",	"Macrophages_LPS 4h_rep1",	"Macrophages_LPS 4h_rep2",	"Cord blood-derived mast cells_control_rep2",	"Neutrophils_unstimulated_rep1",	"Neutrophils_unstimulated_rep2",	"B cells_unstimulated_rep1",	"B cells_unstimulated_rep2",	"Neutrophils_LPS 1h_rep1",	"Effector memory T cells_unstimulated_rep1",	"Effector memory T cells_unstimulated_rep2",	"NK cells_unstimulated_rep1",	"NK cells_unstimulated_rep2",	"Basophils_unstimulated_rep2",	"Central memory T cells_unstimulated_rep1",	"Central memory T cells_unstimulated_rep2",	"Th1 cells_unstimulated_rep1",	"Th1 cells_unstimulated_rep2",	"Th2 cells_unstimulated_rep1",	"Th2 cells_unstimulated_rep2",	"Eosinophils_control_rep2")
celltypes <- c("Eosinophils_control_rep1",	"Eosinophils_PMA 2h_rep1",	"Cord blood-derived mast cells_control_rep1",	"Cord blood-derived mast cells_IgE 2h_rep1",	"Dendritic cells_LPS 48h_rep1",	"Cord blood-derived mast cells_IgE 2h_rep2",	"Immature dendritic cells_rep1",	"Immature dendritic cells_rep2",	"Dendritic cells_LPS 48h_rep2",	"Macrophages_unstimulated_rep1",	"Macrophages_unstimulated_rep2",	"Macrophages_LPS 4h_rep1",	"Macrophages_LPS 4h_rep2",	"Cord blood-derived mast cells_control_rep2",	"Neutrophils_unstimulated_rep1",	"Neutrophils_unstimulated_rep2",	"B cells_unstimulated_rep1",	"B cells_unstimulated_rep2",	"Neutrophils_LPS 1h_rep1",	"Effector memory T cells_unstimulated_rep1",	"Effector memory T cells_unstimulated_rep2",	"NK cells_unstimulated_rep1",	"NK cells_unstimulated_rep2",	"Basophils_unstimulated_rep2",	"Central memory T cells_unstimulated_rep1",	"Central memory T cells_unstimulated_rep2",	"Th1 cells_unstimulated_rep1",	"Th1 cells_unstimulated_rep2",	"Th2 cells_unstimulated_rep1",	"Th2 cells_unstimulated_rep2",	"Eosinophils_control_rep2")
```

```{r fig.height=6, fig.width=8}
garvan_t <- as.data.frame(t(garvan_unique))
colnames(garvan_t) <- my_genes$row.names
garvan_t %<>% mutate(cell_type=celltypes) %>% dselect(cell_type,everything())

write.table(garvan_t, "Garvan_parse_norm.txt", sep="\t", col.names=TRUE, row.names=TRUE)

ggplot(garvan_t, aes(x=reorder(cell_type, -CCR4),y=CCR4)) +
  geom_bar(stat="identity") + # this is reaquired to plot the actual number
  theme(axis.text.x = element_text(angle = 90, hjust = 1,vjust=0.5))

ccr4 <- ggplot(garvan_t, aes(x=cell_type,y=CCR4)) +
  geom_bar(stat="identity") + # this is reaquired to plot the actual number
  theme(axis.text.x = element_text(angle = 90, hjust = 1,vjust=0.5))

ccl22 <- ggplot(garvan_t, aes(x=cell_type,y=CCL22)) +
  geom_bar(stat="identity") + # this is reaquired to plot the actual number
  theme(axis.text.x=element_blank())

ccl17 <- ggplot(garvan_t, aes(x=cell_type,y=CCL17)) +
  geom_bar(stat="identity") + # this is reaquired to plot the actual number
  theme(axis.text.x=element_blank())

foxp3 <- ggplot(garvan_t, aes(x=cell_type,y=FOXP3)) +
  geom_bar(stat="identity") + # this is reaquired to plot the actual number
  theme(axis.text.x=element_blank())

plot_grid(ccl22, ccl17,foxp3,nrow =3)

ggsave("Garvan_Norm_signal2.jpg", dpi=600, width=8, height=5, units="in", plot= last_plot())
```

