---
title: "Mosely & CrownBio lookup"
author: "Mengshu"
date: "7/16/2019"
output: html_document
---

In-house and Mosely microarray expression data for Mouse syngeneic models

mouse_dds is the in-house data of CT26, MC38 and Pan02 in triplicate
mouseData_tum is the Mosely data
samples.info_tum is the metadata for Mosely

```{r}
library(tidyverse)
library(dplyr)
library(magrittr)
library(tibble)
library(ggplot2)
library(data.table)
library(reshape2) #needed for the heatmap
source("/Volumes/Enterprise/FLX/Files_from_Gene/R_functions.r")
#source("D:/FLX/Files_from_Gene/R_functions.r")
```


Load Raw data
```{r}
load("/Volumes/Enterprise/FLX/Silpa_data/mouseAnnot.rdata")
load("/Volumes/Enterprise/FLX/Silpa_data/mouse_dds.rdata") # This is an in-house dataset, not the Mosely data

#This is the Mosely data
load("/Volumes/Enterprise/FLX/Silpa_data/mouseData_tum.rdata")
load("/Volumes/Enterprise/FLX/Silpa_data/samples.info_tum.rdata")

load("D:/FLX/Silpa_data/mouseData_tum.rdata")
load("D:/FLX/Silpa_data/samples.info_tum.rdata")
```

Load Processed Data
```{r}
save(mouse, file="Mosely_merge.rdata")
load("Mosely_merge.rdata")
save(crown_unlog, file="Crown_unlog_unique.rdata")
save(crown_z, file="Crown_z_unique.rdata") #this has been unlogged, betweenarraynormalized, zscored, and unique gened
load("/Volumes/Enterprise/FLX/CrownBio_data/Crown_unlog_unique.rdata")
load("/Volumes/Enterprise/FLX/Reference tables/mouse_human_gene.rdata")
```


Examine data
Plot histogram of signal distribution
```{r}
samples.info_tum
rownames(mouseData_tum) # this is the gene expression data of 24818 genes in columns by 93 rows that are unnamed
mouseData_tum[1:10,1:10]
mouseData_tum %>% gather(Gene,Signal) %>% hist()
mouseData_tum %>% gather(Gene,Signal) %>% ggplot(aes(Signal, colour=Signal)) + geom_histogram(binwidth=0.2) + scale_x_continuous(name="Log2 Normalized Signal (Microarray)", breaks=seq(0,10,by=1)) +
ggsave("Mosely_Signal_Hist.jpg", width=5, height=5, dpi=600, plot= last_plot(), units = "in")

```
Merge the Mosely data and metadata

```{r}

mouse <- data.frame(bind_cols(samples.info_tum,mouseData_tum))
save(mouse,file="Mosely_data_meta.rdata")

mouse %>% dselect(cell.line, genes) %>% gather(gene,RNA_expression, -cell_line) %>% ggplot(aes(x=cell_line, y=RNA_expression, group_by(cell_line), colour= cell_line)) + geom_point(show.legend=FALSE) + facet_wrap(~gene, ncol=1, scales="free_y") + theme(axis.title = element_text(size=14), axis.text.y = element_text(size=14), axis.text.x = element_text(angle=45,size=14,hjust=1),strip.text.x=element_text(size=16)) +
ggsave("GCN2_off_target_Mouse_Vert.jpg", width=6, height=10, dpi=600, plot= last_plot(), units = "in")
names(mouse)[7] <- "cell_line"
mouse %>% dselect(cell_line, genes)  %>% gather(gene,RNA_expression, -cell_line) %>% ggplot(aes(x=cell_line, y=RNA_expression, group_by(cell_line), colour= cell_line)) + geom_point(show.legend=FALSE) + theme(axis.title = element_text(size=14), axis.text.y = element_text(size=14), axis.text.x = element_text(angle=45,size=14,hjust=1)) + stat_summary(inherit.aes = TRUE,quantiles = 0.5, colour="black")
```

Off-target Kinases plotting for Mike Z. 
```{r fig.height=6, fig.width=16}
ordered_cell_lines <- mouse %>% dselect(cell_line, genes) %>% gather(gene,RNA_expression, -cell_line) %>% dfilter(!is.na(RNA_expression)) %>% group_by(cell_line) %>% 
  summarise(value=median(RNA_expression)) %>% arrange(value) %>% pull(cell_line)


mouse %>% dselect(cell_line, genes) %>% gather(gene,RNA_expression, -cell_line) %>% dfilter(!is.na(RNA_expression)) %>% 
  mutate(cell_line = factor(cell_line, levels=ordered_cell_lines)) %>% mutate(gene = factor(gene, levels=c("Cdk6","Cdc7","Mapk8","Mapk9","Gsk3a","Gsk3b","Eif2ak4","Ly6g","gMDSC_score","Trib2"))) %>%
    ggplot(aes(x=cell_line, y=RNA_expression, group=cell_line, colour= cell_line)) + 
  stat_summary(fun.y=median, fun.ymax=median, fun.ymin=median, geom="crossbar", size=0.5, colour="black") + 
  geom_point(show.legend=FALSE) + 
  facet_wrap(~gene, ncol=10, scales= "free_x") + expand_limits(y = c(5,7)) +
  theme(axis.title = element_text(size=14), axis.text.y = element_text(size=12), axis.text.x = element_text(size=10,hjust=1),strip.text.x=element_text(size=14)) +  
  coord_flip() +

  ggsave("GCN2_off_target_Mouse_Horz.jpg", width=16, height=6, dpi=600, plot= last_plot(), units = "in")
```


CrownBio Syngeneic mouse line data
There are a lot of undetected genes, I am going to remove them so that they don't throw off normalization

unlog.ratios: this is the correct function to unlog log2 calculations
unlog.fold.change: does something else I haven't had time to figure out
```{r}
library(limma)
crown <- read.table("C:/Users/mengs/Downloads/RNAseq_expr_syngeneicmodels.csv",sep=",",header=TRUE)
crown <- read.table("/Volumes/Enterprise/FLX/CrownBio_data/RNAseq_expr_syngeneicmodels.csv",sep=",",header=TRUE)

crown %<>% separate(Gene, sep="\\(", into=c("Gene","ENSG")) %>% dselect(-ENSG)

colnames(crown) <- paste0(colnames(crown),"_Crown") # Mark the Crown lines for mergind with Mosely

# Un-log
crown_unlog <- crown %>% mutate_if(is.numeric,unlog.ratios)
#crown_unlog <- crown_unlog[!duplicated(crown_unlog$Gene_Crown),]
#crown_unlog_expressed <- crown_unlog %>% data.frame %>% tibble::column_to_rownames("Gene_Crown") %>% mutate(my_rowsum = crown_unlog %>% rowSums) %>% dfilter(rowsum > )

crown %>% gather(Cell_line,Signal,-Gene_Crown) %>% ggplot(aes(Cell_line, Signal)) + geom_violin(outlier.size=0.1, outlier.alpha = 0.1)

crown_n <- normalizeBetweenArrays(crown_unlog[,2:33]) %>% data.frame
crown_z <- crown_n %>% mutate_if(is.numeric,mx_zscore)

crown_n %>% data.frame %>% gather(Cell_line,Signal) %>% ggplot(aes(Cell_line, Signal)) + geom_violin(outlier.size=0.1, outlier.alpha = 0.1)

crown_z %>% data.frame %>% gather(Cell_line,Signal) %>% ggplot(aes(Cell_line, Signal)) + geom_violin(outlier.size=0.1, outlier.alpha = 0.1)

crown_z %<>% mutate(Gene= crown[,1]) 
crown_z %<>% dselect(Gene, everything())

crown_z <- crown_z[!duplicated(crown_z$Gene),]
```
How many genes in the Crown RNAseq dataset are essentially non-detected??
Histogram of expression signals show a lower limit of -2 on the log2 norm data, which looks like no-detection
```{r}
crown %>% dselect(Gene_Crown,J558_Crown) %>% gather(Cell_line,Signal,-Gene_Crown) %>% ggplot(aes(Cell_line, Signal)) + geom_violin(outlier.size=0.1, outlier.alpha = 0.1)
hist(crown)
crown %>% dfilter(J558_Crown == -2) # 32722 out of 47729 are -2, most of these are not really genes, so that's ok
crown %>% dfilter(J558_Crown != -2) # 15007 transcripts detected
```

Merge Crown and Mosely data
Mosely data has replicates, merge them by cell line, then merge with Crown.
Merge into Mosely because Crown has a lot of non-gene lines that are 0 and useless
Actually, use inner_join, because sva doesn't do NAs well. 
```{r}
boxplot(crown[2:length(colnames(crown))])
crown_unlog # 32 
mouse_merge <- mouse %>% group_by(experimental.group) %>% summarise_if(is.numeric,median) %>% ungroup() %>% tibble::column_to_rownames("experimental.group") %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") # 31 syngeneic lines

#mouse

syngeneic <- inner_join(crown_unlog,mouse_merge, by=c("Gene_Crown"="Gene"))
save(syngeneic,file="Syngeneic_Crown_Mosely.rdata")
boxplot(syngeneic[,2:length(colnames(syngeneic))], log="y")
??graphics::boxplot
syngeneic %>% tibble::column_to_rownames("Gene_Crown") %>% select_if(is.numeric) %>% rowMeans %>% data.frame %>% ggplot(aes(.)) + geom_histogram(binwidth=0.5) + xlim(0,10)
syngeneic %>% select_if(is.numeric) %>% rowSds %>% data.frame %>% ggplot(aes(.)) + geom_histogram(binwidth=0.5) + xlim(0,20)
row_means <- syngeneic %>% tibble::column_to_rownames("Gene_Crown") %>% select_if(is.numeric) %>% rowMeans %>% data.frame
row_sds <- syngeneic %>% tibble::column_to_rownames("Gene_Crown")  %>% select_if(is.numeric) %>% rowSds %>% data.frame
colnames(row_means) <- "r_mean"
colnames(row_sds) <- "r_sd"
syngeneic <- cbind(syngeneic,row_means,row_sds)
syngeneic_f <- syngeneic %>% dfilter(r_mean >= 3, r_sd >= 10)  

```
Batch normalize using sva
There cannot be missing data (NAs) between batches
```{r}
#BiocManager::install("sva")
library(sva)
batches <- as.factor(c(rep(1,32),rep(2,31))) # 1 = Crown 32 lines, 2= Mosely 31 lines
sample <- seq(1,63)
pheno <- data.frame(cbind(sample,batches))
#model_combat <- model.matrix(~batches, data=pheno)
model0 <- model.matrix(~1, data=pheno) # don't include the batch information here in the model, batch info goes only into the ComBat step
model0

Syngeneic_batchcorr <- sva::ComBat(dat=as.matrix(syngeneic[,2:64]), batch=batches, mod=model0, par.prior = TRUE,prior.plots=FALSE)  #, par.prior= TRUE, prior.plots = TRUE 
boxplot(syngeneic_f[,2:60])
boxplot(Syngeneic_batchcorr)
rownames(Syngeneic_batchcorr) <- syngeneic$Gene_Crown
library(limma)
Syngeneic_batchcorr_n <- normalizeBetweenArrays(Syngeneic_batchcorr)
boxplot(Syngeneic_batchcorr_n)
Syngeneic_z <- Syngeneic_batchcorr_n %>% data.frame %>% tibble::rownames_to_column("Gene") %>% mutate_if(is.numeric,mx_zscore) %>% tibble::column_to_rownames("Gene")

```

Cluster cell lines to see if clusters make sense biologically
Rows are samples to cluster, columns are features
hclust(dist(matrix))
```{r fig.height=10, fig.width=10}
syngeneic_hier <- hclust(dist(t(Syngeneic_z_n)))
Syngeneic_z_n <- normalizeBetweenArrays(Syngeneic_z)
plot(syngeneic_hier)

Crown_unlog_n <- normalizeBetweenArrays(crown_unlog %>% group_by(Gene_Crown) %>% summarise_all(mean) %>% ungroup %>% tibble::column_to_rownames("Gene_Crown"))

Crown_hier <- hclust(dist(t(Crown_unlog_n)))
plot(Crown_hier)
```



UMAP cell lines
UMAP
!! UMAP works on the rows of data
To cluster by cell lines, transpose the data to celllines in rows
```{r fig.height=10, fig.width=10}
library(umap)

Syngeneic_umap <- umap(t(Syngeneic_batchcorr_n))
Syngeneic_umap <- umap(t(Syngeneic_z))
save(Syngeneic_umap, file="Syngeneic_umap.rdata")
head(Syngeneic_umap$layout)
Syngeneic_umap_umap_all <- data.frame(Syngeneic_umap$layout,rownames=TRUE)

crown_umap <- Syngeneic_umap_umap_all %>% tibble::rownames_to_column("Line") %>% dfilter(Line %like% "Crown") %>% tibble::column_to_rownames("Line")
CL_umap <- Syngeneic_umap_umap_all %>% tibble::rownames_to_column("Line") %>% dfilter(Line %like% "_CL") %>% tibble::column_to_rownames("Line")
Mosely_T_umap <- Syngeneic_umap_umap_all %>% tibble::rownames_to_column("Line") %>% dfilter(Line %like% "_T") %>% tibble::column_to_rownames("Line")


plot.default(Syngeneic_umap_umap_all, cex= 0.5,pch=20) + #pch are shapes represented by number
points(crown_umap, col="red", pch=20) +
points(CL_umap, col="orange", pch=20) +
points(Mosely_T_umap, col="green", pch=15)

Syngeneic_umap_umap_all %>% tibble::rownames_to_column("Line") %>% mutate(Batch= case_when(Line %like% "_Crown" ~ "Crown", Line %like% "_CL" ~ "Mosely_CL", Line %like% "_T" ~ "Mosely_T")) %>% ggplot(aes(x=X1, y=X2,label = Line, fill= Batch, alpha=0.005)) + geom_label()  + geom_point()
```

Calculate the MDSC scores of each mouse cell line based on our MDSC signatures
Put the score back into the mouse table
```{r}
mouse_gMDSC_genes <- mouseAnnot %>% dfilter(HumanName %in% gMDSC_sig) %>% dselect(GENE.SYMBOL)  # translate the gMDSC signature into mouse gene names
mouse_gMDSC_genes <- mouse_gMDSC_genes[,1] #this makes the data.frame into a simpler list of values
mouse_zscore <- mouse %>% dselect()
  
mx_zscore <- function(x) {result <- (x-mean(x,na.rm=TRUE)) / sd(x,na.rm=TRUE) 
                          return(result)}

mouse_gMDSC_sig <- mouse %>% tibble:: column_to_rownames("SampleID") %>% dplyr::select_if(is.double) %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") %>% mutate_if(is.numeric, mx_zscore) %>% dfilter(Gene %in% mouse_gMDSC_genes) %>% summarize_if(is.numeric, sum)

#GCN2
mouse %>% tibble:: column_to_rownames("SampleID") %>% dplyr::select_if(is.double) %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") %>% mutate_if(is.numeric, mx_zscore) %>% dfilter(Gene =="Eif2ak4") %>% ggplot(aes)

mouse %>% tibble:: column_to_rownames("SampleID") %>% dplyr::select_if(is.double) %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") %>% mutate_if(is.numeric, mx_zscore) %>% gather(Gene,Signal) %>% ggplot(aes(Signal, colour=Signal)) + geom_histogram(binwidth=0.2) + scale_x_continuous(name="Log2 Normalized Signal (Microarray)") +
  ggsave("Mosely_Zscore_Signal_Hist.jpg", width=5, height=5, dpi=600, plot= last_plot(), units = "in")


mouse_gMDSC_sig %>% t %>% data.frame %>% tibble::rownames_to_column("SampleID")
rownames(mouse_gMDSC_sig) <- "gMDSC_score"
#mouse %<>% dselect(-gMDSC_score)
mouse <- left_join(mouse, mouse_gMDSC_sig %>% t %>% data.frame %>% tibble::rownames_to_column("SampleID"), by = "SampleID")
```

ISR response kinase target signature for George
1. Take Z-score of all genes for independent replicates of samples. 
2. Select markers for each Kinase to make a heatmap: examine it to see how specific each marker is for its intended kinase
3. Calculate a weighted score for GCN2-specific ISR marker score

```{r}
mouse_z <- mouse %>% tibble::column_to_rownames("SampleID") %>% dplyr::select_if(is.double) %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") %>% mutate_if(is.numeric, mx_zscore2)
save(mouse_z, file="Mosely_data_zscore.rdata")
load("/Volumes/Enterprise/FLX/General working code/Mosely_data_zscore.rdata") #mouse_z
load("D:/FLX/General working code/Mosely_data_zscore.rdata")
```

ISR signatures
Missing: Cebpb, Ifit3
Lokup mouse gene names
Trfr2 == Tfr2
Stch ==Hspa13
```{r}
mouseAnnot %>% dfilter(HumanName=="TNFRSF12A")

Common <- c("Ddit3","Trib3","Ppp1r15a","Asns")
#GCN2 <- c("Hspa5","Slc3a2","Hspa13","Dct")
PKR <- c("Ifit1","Isg15","Usp18")
#PERK <- c("P4hb", "Xbp1")
PERK_try <- c("Atf3","Tnfrsf12a") #"Try10","Prss2","Sycn","Ccl27b","Timm9","Foxf1","Ccl27a",
#HRI <- c("Adm2","Slc7a11","Grb10","Nupr1","Soat2","Lrrc1","Glipr2","Arl14ep","Pycr1")
HRI_sub <-c("Slc7a11","Soat2","Arl14ep","Pycr1") 
aas_sig_m <- c("Clic4","Yars","Sars","Phgdh","Gars","Asns","Sat1","Slc3a2","Ddit3","Wars","Chac1","Herpud1","Pycr1","Trib3","Cebpb","Ppp1r15a","Atf4") # "C6orf48" is missing and removed

gMDSC_sig <- c("DYSF", "C5AR1", "TREM1", "CSF3R", "DEFA1", "CXCR2", "PLBD1", "CMTM2", "CXCR1", "TNFRSF10C", "LTF", "F13A1", "PPBP", "VNN3", "PADI4", "GLT1D1", "CLEC4D", "LCN2", "BPI", "CAMP", "CD24", "PGLYRP1", "CEACAM1", "S100P", "CYP4F3", "CLC", "S100A12", "MCEMP1", "BST1", "ARG1", "CDA", "ADGRG3", "CSF2RB", "IL1R2", "IL1RAP", "KCNJ15", "LIMK2", "DOCK5", "STX3", "FFAR2", "MEFV", "SIRPB1")
gMDSC_sig_m <- c("Dysf", "C5ar1", "Trem1", "Csf3r", "Defa1", "Cxcr2", "Plbd1", "Cmtm2", "Cxcr1", "Tnfrsf10c", "Ltf", "F13a1", "Ppbp", "Vnn3", "Padi4", "Glt1d1", "Clec4d", "Lcn2", "Bpi", "Camp", "Cd24", "Pglyrp1", "Ceacam1", "S100p", "Cyp4f3", "Clc", "S100a12", "Mcemp1", "Bst1", "Arg1", "Cda", "Adgrg3", "Csf2rb", "Il1r2", "Il1rap", "Kcnj15", "Limk2", "Dock5", "Stx3", "Ffar2", "Mefv", "Sirpb1")
mMDSC_sig_m <- c("Csf3r", "Slc6a6", "Trem1", "Clec4e", "Plbd1")

ISR_markers <- unique(c(Common,PKR,PERK_try,HRI_sub,gMDSC_sig_m, mMDSC_sig_m,"Eif2ak4", "Ido1","Arg1"))
ISR_markers <- c(Common,PKR,PERK_try,HRI_sub,"Eif2ak4", "Ido1","Arg1")

load("D:/FLX/Silpa_data/mouse_human_gene.rdata")
mouseAnnot %>% dfilter(GENE.SYMBOL=="Try10")
```

Mouse z_score histogram
```{r}

mouse_z %>% gather(Cell_line,Signal,-Gene) %>% ggplot(aes(Cell_line, Signal)) + geom_boxplot(outlier.size=0.1, outlier.alpha = 0.1)
```
ISR
gMDSC
Select marker genes out of dataset and merge back with cell line name
```{r fig.height=5, fig.width=3}
load("Mosely_data_meta.rdata")
isr_z <- mouse_z %>% dfilter(Gene %in% ISR_markers)
#gMDSC_z_score <- mouse_z %>% dfilter(Gene %in% gMDSC_sig_m) %>% summarise_if(is.numeric,sum) %>% mutate(Gene="gMDSC_score") %>% dselect(Gene, everything())
#colnames(gMDSC_z_score) <- c("Gene",paste0(mouse$experimental.group,'_',1:93))
#mMDSC_z_score <- mouse_z %>% dfilter(Gene %in% mMDSC_sig_m) %>% summarise_if(is.numeric,sum)
#colnames(mMDSC_z_score) <- c(paste0(mouse$experimental.group,'_',1:93))
#mouse_z %>% dfilter(Gene =="Eif2ak4")
colnames(isr_z) <- c("Gene",paste0(mouse$experimental.group,'_',1:93)) 
 
isr_z %>% mutate(Gene2=Gene) %>% dselect(Gene,Gene2,everything()) %>% arrange(match(Gene,ISR_markers)) %>% write.table(file="ISR_zscore.cdt",row.names = FALSE,sep="\t")

#isr_mdsc_z <- rbind(gMDSC_z_score,isr_z)

#isr_z %>% arrange(match(Gene,ISR_markers)) %>% write.table(file="ISR_zscore.txt",row.names = FALSE,sep="\t")

isr_z %>% gather(Cell,zscore,-Gene) %>% mutate(Gene=factor(Gene,levels=c(rev(ISR_markers)))) %>% ggplot(aes(x=Gene,y=zscore, fill=Gene)) + geom_violin(show.legend=FALSE) + geom_point(size=0.5,show.legend=FALSE) + coord_flip() +  theme(axis.title = element_text(size=8), axis.text.y = element_text(size=16), axis.text.x = element_text(size=16,hjust=1))

crown_z  %>% dfilter(Gene %in% ISR_markers) %>% gather(Cell,zscore,-Gene) %>% mutate(Gene=factor(Gene,levels=c(rev(ISR_markers)))) %>% ggplot(aes(x=Gene,y=zscore, fill=Gene)) + geom_violin(show.legend=FALSE) + geom_point(size=0.5,show.legend=FALSE) + coord_flip() +  theme(axis.title = element_text(size=8), axis.text.y = element_text(size=16), axis.text.x = element_text(size=16,hjust=1)) + ylim(-1,10)
```

Take z-scores of genes across samples: I.e. take per sample cross_gene z-score, then take per gene cross_sample z-score
```{r fig.height=5, fig.width=3}

isr_z_z <- isr_z %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore2) %>% tibble::column_to_rownames("experimental.group") %>% t %>% as.data.frame %>% tibble::rownames_to_column("Gene")

crown_z_z <- crown_z  %>% dfilter(Gene %in% ISR_markers) %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore) %>% tibble::column_to_rownames("experimental.group") %>% t %>% as.data.frame %>% tibble::rownames_to_column("Gene")

isr_z_z %>% gather(Cell,zscore,-Gene) %>% mutate(Gene=factor(Gene,levels=rev(ISR_markers))) %>% ggplot(aes(x=Gene,y=zscore, fill=Gene)) + geom_violin(show.legend=FALSE) + geom_point(size=0.5,show.legend=FALSE) + coord_flip() +  theme(axis.title = element_text(size=8), axis.text.y = element_text(size=16), axis.text.x = element_text(size=16,hjust=1))

crown_z_z %>% gather(Cell,zscore,-Gene) %>% mutate(Gene=factor(Gene,levels=rev(ISR_markers))) %>% ggplot(aes(x=Gene,y=zscore, fill=Gene)) + geom_violin(show.legend=FALSE) + geom_point(size=0.5,show.legend=FALSE) + coord_flip() +  theme(axis.title = element_text(size=8), axis.text.y = element_text(size=16), axis.text.x = element_text(size=16,hjust=1))

isr_z %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore2) %>% tibble::column_to_rownames("experimental.group") %>% t %>% as.data.frame %>% tibble::rownames_to_column("Gene") %>% mutate(Gene2=Gene) %>% dselect(Gene,Gene2,everything()) %>% write.table(file="ISR_zscore_twice.cdt",row.names = FALSE,sep="\t")

crown_z %>% dfilter(Gene %in% ISR_markers) %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore) %>% tibble::column_to_rownames("experimental.group") %>% t %>% data.frame %>% tibble::rownames_to_column("Gene") %>% dselect(Gene,everything()) %>% write.table(file="Crown_zscore_twice.cdt",row.names =FALSE,sep="\t")
```

Correlation of markers
```{r fig.height=5, fig.width=5}
  get_lower_tri<-function(cormat){
    cormat[upper.tri(cormat)] <- NA
    return(cormat)
  }

  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }

isr_genes_z <- isr_z %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame()

isr_genes_zz <- isr_z %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore2) %>% tibble::column_to_rownames("experimental.group")  

isr_genes_zz <- crown_z %>% dfilter(Gene %in% ISR_markers) %>% arrange(match(Gene,ISR_markers)) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("experimental.group") %>% mutate_if(is.numeric,mx_zscore) %>% tibble::column_to_rownames("experimental.group")  

med_sig_cor <- round(cor(isr_genes_zz),1)
med_sig_cor <- get_upper_tri(med_sig_cor)
melt_med_sig_cor <- melt(med_sig_cor)
melt_med_sig_cor %>% ggplot(aes(x=Var1, y=Var2, fill=value, label=value)) + geom_tile(color = "white")+
    geom_text(aes(Var1, Var2, label = value), color = "grey30", size = 2) +
 scale_fill_gradient2(low = "steelblue", high = "red", na.value="white", limit = c(-0.5,1), space = "Lab", 
   name="Pearson\nCorrelation") +
  theme_minimal()+ 
    scale_x_discrete(position = "top") +
 theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 0))+
 coord_fixed() +
ggsave("Mosely_Sig_corr_median_heatmap_Sep21.jpg", width=5, height=5, dpi=600, plot= last_plot(), units = "in")  

HRI_sub <-c("Slc7a11","Soat2","Arl14ep","Pycr1") 

``` 



Calculate GCN2 signature from other ISR signatures
```{r}

Common_s <- isr_z_z %>% dfilter(Gene %in% Common) %>% summarise_if(is.numeric,sum)
PKR_s <- isr_z_z %>% dfilter(Gene %in% PKR) %>% summarise_if(is.numeric,sum)
PERK_s <- isr_z_z %>% dfilter(Gene %in% PERK_try) %>% summarise_if(is.numeric,sum)
HRI_s <- isr_z_z %>% dfilter(Gene %in% HRI_sub) %>% summarise_if(is.numeric,sum)
GCN2_s <- isr_z_z %>% dfilter(Gene %in% GCN2) %>% summarise_if(is.numeric,sum)
AAS_s <- isr_z_z %>% dfilter(Gene %in% aas_sig_m) %>% summarise_if(is.numeric,sum)
gMDSC_s <- mouse_z %>% dfilter(Gene %in% gMDSC_sig_m) %>% summarise_if(is.numeric,sum) # %>% mutate(Gene="gMDSC_score") %>% dselect(Gene, everything())
colnames(gMDSC_s) <- c(paste0(mouse$experimental.group,'_',1:93))
mMDSC_s <- mouse_z %>% dfilter(Gene %in% mMDSC_sig_m) %>% summarise_if(is.numeric,sum)
colnames(mMDSC_s) <- c(paste0(mouse$experimental.group,'_',1:93))

Common_s <- isr_z %>% dfilter(Gene %in% Common) %>% summarise_if(is.numeric,sum)
PKR_s <- isr_z %>% dfilter(Gene %in% PKR) %>% summarise_if(is.numeric,sum)
PERK_s <- isr_z %>% dfilter(Gene %in% PERK) %>% summarise_if(is.numeric,sum)
HRI_s <- isr_z %>% dfilter(Gene %in% HRI_sub) %>% summarise_if(is.numeric,sum)
GCN2_s <- isr_z %>% dfilter(Gene %in% GCN2) %>% summarise_if(is.numeric,sum)

#CN2_s <- (Common_s - PKR_s - PERK_s - HRI_s)

ISR_s <- bind_rows(Common_s,PKR_s,PERK_s,HRI_s,GCN2_s,AAS_s,gMDSC_s,mMDSC_s)
rownames(ISR_s) <- c("Common","PKR","PERK","HRI","GCN2s","AAS","gMDSC","mMDSC")
ISR_s_t <- ISR_s %>% t %>% data.frame 
ISR_s_t <- bind_cols(samples.info_tum %>% dselect(`experimental group`),ISR_s_t)

ISR_median <- ISR_s_t %>% mutate(PKRi= ifelse(PKR<0,0,PKR), PERKi=ifelse(PERK<0,0,PERK), HRIi=ifelse(HRI<0,0,HRI)) %>% 
  mutate(GCN2_specific = Common + GCN2s - PKRi - PERKi - HRIi) %>% mutate(Group=`experimental group`) %>% dfilter(Group %like% "_T$") %>% group_by(Group) %>% summarise_if(is.double,median) %>% ungroup  %>% mutate(Group2=Group) %>% dselect(Group,Common,GCN2s,PKR,PERK,HRI,GCN2_specific,AAS,mMDSC,gMDSC) %>% arrange(desc(GCN2_specific)) %>% ungroup()

ISR_median %>% write.table(file="ISR_scores_grouped_Sep18_New.cdt",row.names =FALSE,sep="\t",quote=FALSE)

```

Calculate AAS scores
```{r}

aas_sig <- c("CLIC4","YARS","SARS","PHGDH","C6orf48","GARS","ASNS","SAT1","SLC3A2","DDIT3","WARS","CHAC1","HERPUD1","PYCR1","TRIB3","CEBPB","PPP1R15A","ATF4")

aas_sig_m <- c("Clic4","Yars","Sars","Phgdh","Gars","Asns","Sat1","Slc3a2","Ddit3","Wars","Chac1","Herpud1","Pycr1","Trib3","Cebpb","Ppp1r15a","Atf4") # "C6orf48" is missing and removed

 %>% dfilter(GENE.SYMBOL %in% aas_sig_m)
```
Signal Scores from Complete CrownBio data
```{r}
Common_s <- crown_z_z %>% dfilter(Gene %in% Common) %>% summarise_if(is.numeric,sum)
PKR_s <- crown_z_z %>% dfilter(Gene %in% PKR) %>% summarise_if(is.numeric,sum)
PERK_s <- crown_z_z %>% dfilter(Gene %in% PERK_try) %>% summarise_if(is.numeric,sum)
HRI_s <- crown_z_z %>% dfilter(Gene %in% HRI_sub) %>% summarise_if(is.numeric,sum)
GCN2_s <- crown_z_z %>% dfilter(Gene %in% GCN2) %>% summarise_if(is.numeric,sum)
AAS_s <- crown_z_z %>% dfilter(Gene %in% aas_sig_m) %>% summarise_if(is.numeric,sum)
gMDSC_s <- crown_z_z %>% dfilter(Gene %in% gMDSC_sig_m) %>% summarise_if(is.numeric,sum)
mMDSC_s <- crown_z_z %>% dfilter(Gene %in% mMDSC_sig_m) %>% summarise_if(is.numeric,sum)
crown_z_z %>% dfilter(Gene =="Csf3r")
ISR_s <- bind_rows(Common_s,PKR_s,PERK_s,HRI_s,GCN2_s,AAS_s,gMDSC_s,mMDSC_s)
rownames(ISR_s) <- c("Common","PKR","PERK","HRI","GCN2s","AAS","gMDSC","mMDSC")
ISR_s_t <- ISR_s %>% t %>% data.frame 
#ISR_s_t <- bind_cols(samples.info_tum,ISR_s_t)
lines <- rownames(ISR_s_t)
ISR_s_t %>% mutate(PKRi= ifelse(PKR<0,0,PKR), PERKi=ifelse(PERK<0,0,PERK), HRIi=ifelse(HRI<0,0,HRI)) %>% mutate(GCN2_specific = Common - PKRi - PERKi - HRIi)  %>% arrange(desc(GCN2_specific)) %>% mutate(Cell_line=lines, Cellline=lines) %>% dselect(-PKRi,-PERKi, -HRIi) %>% dselect(Cell_line,Cellline,GCN2s,Common,PKR,PERK,HRI,GCN2_specific,AAS,gMDSC,mMDSC) %>% as.data.table %>% write.table(file="ISR_scores_grouped_ZZscore_CrownBio_GCN2_s.cdt",row.names = FALSE,sep="\t")


```



Manually Downloaded CrownBio RNA signal
```{r}
crownbio <- read.table("D:/FLX/General\ working\ code/Mouse_syngeneic_RNA_CrownBio.txt",sep="\t",header=TRUE)
CrownBio_RNA <- crownbio %>% dselect(-CANCER.TYPE,-ID) %>% mutate() %>% spread(GENE,LOG2.FPKM.) %>% mutate(GENE=factor(GENE,levels=rev(ISR_markers)))
gene_order <- c("Clic4","Yars","Sars","Phgdh","Gars","Sat1","Slc3a2","Wars","Chac1","Herpud1","Pycr1","Cebpb","Atf4","Asns","Ddit3","Trib3","Ppp1r15a","Hspa5","Slc3a2","Hspa13","Ifit1","Isg15","Usp18","P4hb","Slc7a11","Soat2","Arl14ep")
CrownBio_RNA_Z <- CrownBio_RNA %>% mutate_if(is.numeric,mx_zscore)

CrownBio_RNA %>% mutate_if(is.numeric,mx_zscore) %>% mutate(Sample=SAMPLE.NAME) %>% select(SAMPLE.NAME, Sample, everything()) %>% arrange(desc(ASNS)) %>%  write.table(file="CrownBio_RNA_Z.txt", quote=FALSE, sep="\t", row.names = FALSE)


Common <- c("Ddit3","Trib3","Ppp1r15a","Asns")
GCN2 <- c("Hspa5","Slc3a2","Hspa13")
PKR <- c("Ifit1","Isg15","Usp18")
PERK <- c("P4hb")
HRI <- c("Adm2","Slc7a11","Grb10","Nupr1","Soat2","Lrrc1","Glipr2","Arl14ep","Pycr1")
HRI_sub <-c("Slc7a11","Soat2","Arl14ep","Pycr1") 
aas_sig_m <- c("Clic4","Yars","Sars","Phgdh","Gars","Asns","Sat1","Slc3a2","Ddit3","Wars","Chac1","Herpud1","Pycr1","Trib3","Cebpb","Ppp1r15a","Atf4")
```

