---
title: "Plot_CCLE_TCGA_CPTAC_ImmProt_MX_112723.Rmd"
author: "Mengshu Xu"
date: "11/27/2023"
output: html_document
---

```{r}
library(magrittr)
library(ggplot2)
library(stringr)
library(dplyr)
library(tidyr)
library(ggchicklet)
library(Hmisc)
library(ggpubr)
library(ggbeeswarm) # for ggplot function geom_quasirandom
library(RColorBrewer)
library(EnvStats)
library(ggpubr)
library(pheatmap)
library(survminer)
library(forcats)
```


Markers
```{r}
my_marker <- c("SIGLEC10","CLEC4E","STAB1","HCK","SIGLEC15")
my_marker <- c("CD5","UBASH3A","UBASH3B")
```

Plot CCLE mRNA expression

```{r fig.height=5, fig.width=16}
#Load CCLE data
load(file="/fcrbiouatappn01/resbioinfo/data/Oncology/CCLE_Data_Formatted/Formatted.CCLE.Data.Rdata")

ccle_rna <- data.frame(rna.data) %>% tibble::rownames_to_column("Gene") # convert the RNA data matrix to a data.frame and create column "Gene" from the rownames

#pull out marker genes of interestd
HHLA_ccle_rna <- ccle_rna %>% filter(Gene %in% my_marker) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample.ID") %>% gather(SLC,mRNA_Log2TPM,-Sample.ID)

#pull out useful meta data from the meta data matrix file
CCLE_meta <- cell.line.info %>% mutate(Cell.Line.ID=str_replace_all(Cell.Line.ID,"-",".")) %>% mutate(Sample.ID=Cell.Line.ID) %>% dplyr::select(Sample.ID, Lineage.Subtype,Lineage.Sub.Subtype,Primary.Disease)

#merge meta data with expression data
HHLA_ccle_rna %<>% left_join(CCLE_meta, by="Sample.ID")

#plot
HHLA_ccle_rna %>% filter(SLC=="CD5") %>% group_by(Primary.Disease) %>% mutate(Cell_lines=n()) %>% ungroup %>% filter(Cell_lines >=30) %>%
  ggplot(aes(x=Primary.Disease, y=mRNA_Log2TPM, color=Primary.Disease)) +
  geom_quasirandom(show.legend = F) +
  facet_wrap(~SLC,ncol=1) +
  theme_bw() +
  stat_summary(fun="median", geom="crossbar",color="grey50") +
  #stat_n_text(text.box=F, size=4.5) +
  theme(axis.text.x=element_text(angle=35,size=14,hjust=1),strip.text=element_text(size=11)) +
  theme(plot.margin=unit(c(0.5,0.5,0.5,2),"cm")) 
```

Plot 2D scatter plot between 2 markers

PDCD1 is a T cell marker sho should not be plotted from CCLE data
```{r fig.height=15, fig.width=15}
HHLA_ccle_rna <- ccle_rna %>% filter(Gene %in% c("HHLA2","TMIGD2","KIR3DL3","CD274","PDCD1","CD3E","CD8A")) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample.ID")
HHLA_ccle_rna %<>% left_join(CCLE_meta, by="Sample.ID")

HHLA_ccle_rna %>% 
  ggplot(aes(x=PDCD1,y=TMIGD2,colour=Primary.Disease)) +
  geom_point() +
  theme_light() +
  facet_wrap(~Primary.Disease,scales="free") +
  NoLegend()

HHLA_ccle_rna %>% 
  ggplot(aes(x=CD3E,y=CD8A,colour=Primary.Disease)) +
  geom_point() +
  theme_light() +
  facet_wrap(~Primary.Disease,scales="free") +
  NoLegend()
```





Plot CCLE DepMap Gene Dependency data

Load 2022Q4 Data Release
```{r}
#gene_dependency <- read.table(file="/fcrbiouatappn01/home/mxu4/reference/Achilles_DepMap_22Q4/CRISPRGeneDependency.csv", sep=",",header=T)

#my_cols <- data.frame(genes=as.character(colnames(gene_dependency))) %>% separate(genes, into=c("Gene",NA), sep="\\.") %>% pull(Gene)
#colnames(gene_dependency) <- my_cols
#save(gene_dependency, file="/fcrbiouatappn01/export/home/mxu4/reference/Achilles_DepMap_22Q4/CRISPRGeneDependency.rdata")

gene_effect <- read.table(file="/fcrbiouatappn01/home/mxu4/reference/Achilles_DepMap_22Q4/CRISPRGeneEffect.csv", sep=",",header=T)

my_cols <- data.frame(genes=as.character(colnames(gene_effect))) %>% separate(genes, into=c("Gene",NA), sep="\\.") %>% pull(Gene)
colnames(gene_effect) <- c("ModelID",my_cols[2:17454])
#gene_effect %<>% dplyr::rename(ModelID="X")
save(gene_effect, file="/fcrbiouatappn01/export/home/mxu4/reference/Achilles_DepMap_22Q4/CRISPRGeneEffect.rdata")

#DATA is HERE
load("/fcrbiouatappn01/export/home/mxu4/reference/Achilles_DepMap_22Q4/CRISPRGeneEffect.rdata")
gene_effect
```

Load CCLE metadata
```{r}
load(file="/fcrbiouatappn01/resbioinfo/data/Oncology/CCLE_Data_Formatted/Formatted.CCLE.Data.Rdata")

my_info <- cell.line.info %>% dplyr::select(Cell.Line.ID,Cell.Line.Name.Stripped,Sample.Collection.Site,Primary.Metastasis,Primary.Disease,Disease.Subtype,Tissue.Lineage,Plot.Order) %>% mutate(Cell.Line.ID=str_replace_all(Cell.Line.ID,"-",".")) %>% tibble::rownames_to_column("DepMap_ID")

```

Merge Data
```{r fig.height=6, fig.width=5}
depmap_meta <- gene_effect %>% dplyr::select(ModelID,my_marker) %>% left_join(my_info,by=c("ModelID"="DepMap_ID")) 

depmap_meta %>% mutate(Tissue.Lineage=fct_reorder(Tissue.Lineage,-UBASH3A)) %>%
  ggplot(aes(x=Tissue.Lineage,y=UBASH3A)) +
  geom_point(size=4, alpha=0.2) +
  theme_bw() +
  coord_flip() +
  geom_hline(yintercept = 0, size=0.5,lty="dashed") +
  geom_hline(yintercept = -1, size=0.5,lty="dashed", color="red3") +
  ylab("UBASH3A DepMap CRISPR Dependency Score 22Q4")

#Compound plot
depmap_meta %>% select(my_marker,Tissue.Lineage) %>% gather(Marker,Effect,-Tissue.Lineage) %>%
  ggplot(aes(x=Tissue.Lineage,y=Effect)) +
  geom_point(size=4, alpha=0.2) +
  theme_bw() +
  facet_wrap(~Marker,ncol=5) +
  coord_flip() +
  geom_hline(yintercept = 0, size=0.5,lty="dashed") +
  geom_hline(yintercept = -1, size=0.5,lty="dashed", color="red3") +
  ylab("DepMap CRISPR Dependency Score 22Q4")
```
Merge Data IL18RAP
```{r fig.height=6, fig.width=6}
depmap_meta <- gene_effect %>% select(ModelID,IL18RAP) %>% left_join(my_info,by=c("ModelID"="DepMap_ID")) 

depmap_meta %>% mutate(Tissue.Lineage=fct_reorder(Tissue.Lineage,-IL18RAP)) %>%
  ggplot(aes(x=Tissue.Lineage,y=IL18RAP)) +
  geom_point(size=4, alpha=0.2) +
  theme_bw() +
  coord_flip() +
  geom_hline(yintercept = 0, size=0.5,lty="dashed") +
  geom_hline(yintercept = -1, size=0.5,lty="dashed", color="red3") +
  ylab("IL18RAP DepMap CRISPR Dependency Score 22Q4")
```

TCGA Expression

TCGA expression
This data has separate values for different sample types!
 01A  01B  01C  01R  02A  02B  03A  03B  05A  06A  06B  07A  11A  11B  11C 
9572  129    4    1   40    6  158   15   11  393    2    1  719   17    1 

11 is adjacent normal

```{r}
# Parsed RNAseq data
load(file="/fcrbiouatappn01/home/mxu4/TCGA/EBPlusPlusAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp_vst_df_MX.rdata") # RNAseq_vst_df Gene
RNAseq_vst_df # The patient IDs are in this format: "TCGA.OR.A5J1.01A.11R.A29S.07" I do need to change and shorten it to join

str(RNAseq_vst_df)

my_marker <- c("SLC7A11","PVRL4","DSG3")
my_marker <- c("CD3E","DGKA")
my_marker <- c("CD200","CD200R1","CD3E","PTPRC","CD8A","CD19","CD79A")
my_marker <- c("B3GNT2","CD8A","CD3E")
my_marker <- c("HHLA2","TMIGD2","KIR3DL3","CD274","PDCD1","CD3E","CD8A")
my_marker <- c("PDCD1","CXCR3","CD8A","CD3E")

SLC_TCGA <- RNAseq_vst_df %>% filter(Gene %in% my_marker) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample.ID") %>% mutate(Sample.ID= str_replace_all(Sample.ID,"\\.","-")) %>% mutate(Sample.ID = substr(Sample.ID,1,16)) %>% mutate(Sample_type= substr(Sample.ID,14,16)) %>% select(Sample.ID,Sample_type,everything()) %>% mutate(Tissue=if_else(grepl("11",Sample_type),"Normal","Tumor")) %>% mutate(Tissue=factor(Tissue,levels=c("Tumor","Normal"))) %>% select(-Sample_type) %>% mutate(Sample.ID = substr(Sample.ID,1,12))

#TCGA Metadata
load(file="/fcrbiouatappn01/home/mxu4/TCGA/TCGA_ImmuneLandscapeOfCancer_Tumors_ImmuneScores.rdata") # TCGA_TMB_IOscore
TCGA_TMB_IOscore

#Merge meta data with mRNA expression data for plotting
SLC_TCGA_meta <- SLC_TCGA %>% left_join(TCGA_TMB_IOscore %>% select(TCGA_Participant_Barcode,TCGA_Study,TCGA_Subtype,OS_Time,OS), by=c("Sample.ID"="TCGA_Participant_Barcode")) %>% filter(!is.na(TCGA_Study))

SLC_TCGA_meta %<>% select(my_marker,TCGA_Study,Tissue) %>% gather(Marker,Log2_TPM,-TCGA_Study,-Tissue)
```


Plot TCGA for SLC
 # 12
```{r fig.height=4, fig.width=18}
library(ggbeeswarm)
library(EnvStats)
tumor_order <- SLC_TCGA_meta %>% filter(Marker=="CLEC4E") %>% mutate(TCGA_Study= fct_reorder(TCGA_Study,-Log2_TPM)) %>% arrange(TCGA_Study) %>% select(TCGA_Study) %>% unique %>% pull(TCGA_Study) %>% as.character()

SLC_TCGA_meta %>% mutate(TCGA_Study= fct_reorder(TCGA_Study,-Log2_TPM)) %>%
  ggplot(aes(x=TCGA_Study, y=Log2_TPM,color=Tissue)) +
  geom_quasirandom(dodge.width = 0.8, show.legend = T, size=0.5) +
  theme_bw() +
  facet_wrap(~Marker,ncol=1) +
  stat_summary(data=subset(SLC_TCGA_meta, Tissue=="Tumor",Marker=="CD5"),fun="median", geom="crossbar",color="grey50", width=0.75, position=position_nudge(x = -0.2, y = 0)) +
  stat_n_text(text.box=F) +
  theme(axis.text.x=element_text(angle=35,size=11,hjust=1),strip.text=element_text(size=11)) +
  ylab("Log2(TPM)")

#FACET
library(tidytext)

SLC_TCGA_meta %>%
  ggplot(aes(x=reorder_within(TCGA_Study,-Log2_TPM,Marker,sep="."),y=Log2_TPM,color=Tissue)) +
  geom_quasirandom(dodge.width = 0.8, show.legend = T, size=0.5) +
  theme_bw() +
  facet_wrap(~Marker,ncol=1, scales="free_x") +
  stat_summary(data=subset(SLC_TCGA_meta, Tissue=="Tumor"),fun="median", geom="crossbar",color="grey50", width=0.75, position=position_nudge(x = -0.2, y = 0)) +
  stat_n_text(text.box=F) +
  theme(axis.text.x=element_text(angle=35,size=11,hjust=1),strip.text=element_text(size=11)) +
  ylab("Log2(TPM)")

```

Correlate Marker with Tumor Purity or Immune Infiltration level
Load TCGA

```{r}
# Parsed RNAseq data
load(file="/fcrbiouatappn01/home/mxu4/TCGA/EBPlusPlusAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp_vst_df_MX.rdata") # RNAseq_vst_df Gene
RNAseq_vst_df # The patient IDs are in this format: "TCGA.OR.A5J1.01A.11R.A29S.07" I do need to change and shorten it to join

Marker_TCGA <- RNAseq_vst_df %>% filter(Gene %in% c("B3GNT2","CD3E","CD8A")) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample.ID") %>% mutate(Sample.ID= str_replace_all(Sample.ID,"\\.","-")) %>% mutate(Sample.ID = substr(Sample.ID,1,16)) %>% mutate(Sample_type= substr(Sample.ID,14,16)) %>% select(Sample.ID,Sample_type,everything()) %>% mutate(Tissue=if_else(grepl("11",Sample_type),"Normal","Tumor")) %>% select(-Sample_type) %>% mutate(Sample.ID = substr(Sample.ID,1,12))

#TCGA Metadata
load(file="/fcrbiouatappn01/home/mxu4/TCGA/TCGA_ImmuneLandscapeOfCancer_Tumors_ImmuneScores.rdata") # TCGA_TMB_IOscore
TCGA_TMB_IOscore
```

Load Tumor Purity Estimate
```{r}
estimate_tcga <- readxl::read_xlsx("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/TCGA_PanCancer/TCGAbiolinks_TumorPurity.xlsx",sheet="Purity")  %>% mutate(Sample.ID = PatientID)
# %>% filter(`Cancer type`=="PRAD")

estimate_hnsc <- readxl::read_xlsx("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/TCGA_PanCancer/TCGAbiolinks_TumorPurity.xlsx",sheet="Purity") %>% filter(`Cancer type`=="HNSC") %>% mutate(Sample.ID = PatientID)

estimate_skcm <- readxl::read_xlsx("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/TCGA_PanCancer/TCGAbiolinks_TumorPurity.xlsx",sheet="Purity") %>% filter(`Cancer type`=="SKCM") %>% mutate(Sample.ID = PatientID)

estimate_lihc <- readxl::read_xlsx("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/TCGA_PanCancer/TCGAbiolinks_TumorPurity.xlsx",sheet="Purity") %>% filter(`Cancer type`=="LIHC") %>% mutate(Sample.ID = PatientID)

estimate_prad <- readxl::read_xlsx("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/TCGA_PanCancer/TCGAbiolinks_TumorPurity.xlsx",sheet="Purity") %>% filter(`Cancer type`=="PRAD") %>% mutate(Sample.ID = PatientID)
```

Merge mRNA with purity data

how is the DGKA split on HNSC KM analysis? 317 high  182 low 499 total  577
```{r fig.height=12, fig.width=16}
# find where is the cutoff point Cor DGKA KM analysis

Marker_estimate <- Marker_TCGA %>% select(Sample.ID,B3GNT2) %>% right_join(estimate_tcga %>% select(-`Sample ID`,-Barcode,-PatientID), by="Sample.ID") %>% group_by(`Cancer type`) %>% dplyr::summarize(Cor=cor(B3GNT2,CPE,method="pearson",use = "complete.obs"),Mean=mean(B3GNT2,na.rm=T))

#Facet
SLC_TCGA_meta %>% filter(Indication_size >=200) %>%
  ggplot(aes(x=B3GNT2,y=Tumor_Fraction)) + #ABSOLUTE, LUMP, ESTIMATE, IHC,  CPE
  geom_point(alpha=0.5) +
  theme_pubr() +
  facet_wrap(~TCGA_Study) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  ylim(NA,1.2) +
  scale_color_gradient(low="grey",high="red") +
  ylab("Combined Tumor Purity Score")

DGKA_estimate %>%
  ggplot(aes(x=DGKA,y=CD3E, outline=NA)) + #ABSOLUTE, LUMP, ESTIMATE, IHC,  CPE
  geom_point() +
  theme_bw() +
  stat_cor(method="pearson")


```

#Plot the Mean CD200 expression with correlation of CD200 to Tumor purity
```{r fig.height=5, fig.width=5}
Marker_TCGA %>% select(Sample.ID,B3GNT2) %>% right_join(estimate_tcga %>% select(-`Sample ID`,-Barcode,-PatientID), by="Sample.ID") %>% group_by(`Cancer type`) %>% dplyr::summarize(Cor_B3GNT2_Purity=cor(B3GNT2,CPE,method="pearson",use = "complete.obs"),Mean_B3GNT2=mean(B3GNT2,na.rm=T)) %>%
  ggplot(aes(x=Cor_B3GNT2_Purity,y=Mean_B3GNT2)) +
  geom_text(aes(label=`Cancer type`)) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  theme_bw()
```
Merge mRNA data with Tumor meta data but not Estimate data
```{r fig.height=12, fig.width=24}
SLC_TCGA_meta <- SLC_TCGA %>% left_join(TCGA_TMB_IOscore, by=c("Sample.ID"="TCGA_Participant_Barcode")) 

#x=HHLA2,y=CD274 no correlation

SLC_TCGA_meta%>% 
  ggplot(aes(x=HHLA2,y=PDCD1)) +
  geom_point() +
  theme_light() +
  facet_wrap(~TCGA_Study,scales="free") +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  NoLegend()

SLC_TCGA_meta %>% 
  ggplot(aes(x=TMIGD2,y=CD274)) +
  geom_point() +
  theme_bw() +
  facet_wrap(~TCGA_Study,scales="free",ncol=8) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  NoLegend()
```



Merge mRNA data with Tumor meta data
```{r}
SLC_TCGA_meta <- SLC_TCGA %>% left_join(TCGA_TMB_IOscore, by=c("Sample.ID"="TCGA_Participant_Barcode")) %>% left_join(estimate_tcga %>% dplyr::rename(Tumor_Fraction="CPE") %>% select(Sample.ID,Tumor_Fraction), by="Sample.ID") %>% group_by(TCGA_Study) %>% mutate(Indication_size=n()) %>% ungroup 
```

Plot CD200 correlation with CD8 and CD19
```{r fig.height=10, fig.width=16}
SLC_TCGA_meta %>% filter(Indication_size >=200) %>%
  ggplot(aes(x=B3GNT2,y=CD3E)) + #ABSOLUTE, LUMP, ESTIMATE, IHC,  CPE
  geom_point(alpha=0.5) +
  theme_pubr() +
  facet_wrap(~TCGA_Study) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  scale_color_gradient(low="grey",high="red") 

SLC_TCGA_meta %>% filter(Indication_size >=200) %>%
  ggplot(aes(x=CD200R1,y=CD79A)) + #ABSOLUTE, LUMP, ESTIMATE, IHC,  CPE
  geom_point(alpha=0.5) +
  theme_pubr() +
  facet_wrap(~TCGA_Study) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  scale_color_gradient(low="grey",high="red") 
```
Correlate CD200 with CD200 correlation with CD8A
```{r fig.height=5, fig.width=5}
SLC_TCGA_meta %>% group_by(TCGA_Study) %>% dplyr::summarize(Cor_CD200_CD79A=cor(CD200,CD79A,method="pearson",use = "complete.obs"),Mean_CD200=mean(CD200,na.rm=T)) %>%
  ggplot(aes(x=Cor_CD200_CD79A,y=Mean_CD200)) +
  geom_text(aes(label=TCGA_Study)) +
  stat_cor(method="pearson") +
  geom_smooth(method="lm") +
  theme_bw()
```
TCGA Correlate Marker with All TCGA Metadata
```{r fig.height=26, fig.width=12}
library(purrr)
library(tidyr)
library(Hmisc)

#Look for indications with enough n, filter for
cancer_types <- SLC_TCGA_meta %>% filter(Indication_size >=100,!is.na(TCGA_Study)) %>% select(TCGA_Study) %>% unique %>% pull(TCGA_Study)

#Create a column that is B3GNT2 normalized to CD3E
SLC_TCGA_meta %<>% mutate(B3GNT2_norm_CD3E=B3GNT2/CD3E)
#make a function to cycle thru different cancers with sufficient data
cor_cancer <- function(x) {
  cor_input <- SLC_TCGA_meta %>% filter(TCGA_Study == x) %>% select(-TCGA_Study,-Indication_size) %>% select_if(is.numeric) %>% as.matrix
  return(rcorr(cor_input)) 
}
cancer_cor_results <- lapply(cancer_types, cor_cancer)
names(cancer_cor_results) <- cancer_types

cancer_corr_CD200 <- cbind.data.frame(lapply(cancer_cor_results,function(j) {j$r %>% data.frame %>% select(CD200)}))
colnames(cancer_corr_CD200) <- cancer_types

cancer_corr_CD200[is.na(cancer_corr_CD200)] <- 0

cancer_corr_CD200R1 <- cbind.data.frame(lapply(cancer_cor_results,function(j) {j$r %>% data.frame %>% select(CD200R1)}))
colnames(cancer_corr_CD200R1) <- cancer_types

cancer_corr_CD200R1[is.na(cancer_corr_CD200R1)] <- 0


cancer_corr_B3GNT2 <- cbind.data.frame(lapply(cancer_cor_results,function(j) {j$r %>% data.frame %>% select(B3GNT2)}))
colnames(cancer_corr_B3GNT2) <- cancer_types

cancer_corr_B3GNT2[is.na(cancer_corr_B3GNT2)] <- 0

cancer_corr_B3GNT2_norm <- cbind.data.frame(lapply(cancer_cor_results,function(j) {j$r %>% data.frame %>% select(B3GNT2_norm_CD3E)}))
colnames(cancer_corr_B3GNT2_norm) <- cancer_types

cancer_corr_B3GNT2[is.na(cancer_corr_B3GNT2_norm)] <- 0

#get P values
cancer_corr_Pval_CD200 <- cbind.data.frame(lapply(cancer_cor_results,function(j) {j$P %>% data.frame %>% select(CD200)}))
colnames(cancer_corr_Pval_CD200) <- cancer_types

#Filter for the most happening signatures
cancer_corr_CD200_filter <- cancer_corr_CD200 %>% filter(PAAD >=0.2 | PAAD <=-0.2 | LUAD <=-0.2  )
cancer_corr_CD200R1_CD200 <- cancer_corr_CD200R1["CD200",]

#Filter for CD200 Correlation with CD200R1
cancer_corr_CD200R1_CD200 <- cancer_corr_CD200R1 %>% filter(PAAD >=0.2 | PAAD <=-0.2 | LUAD <=-0.2  )

pheatmap(cancer_corr_CD200, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)
pheatmap(cancer_corr_CD200_filter, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)

pheatmap(cancer_corr_CD200R1, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)
pheatmap(cancer_corr_CD200R1_filter, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)



pheatmap(cancer_corr_CD200R1_CD200, color= myColor, breaks=myBreaks,cellwidth = 25,cellheight = 25,cluster_cols = F,cluster_rows=F)

pheatmap(cancer_corr_B3GNT2, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)
pheatmap(cancer_corr_B3GNT2_norm, color= myColor, breaks=myBreaks,cellwidth = 15,cellheight = 15)

paletteLength <- 50
myColor <- colorRampPalette(c("blue", "white", "red"))(paletteLength)


myBreaks <- c(seq( -1, 0, length.out=ceiling(paletteLength/2) + 1), 
              seq( 1/50,1,length.out=floor(paletteLength/2))  ) 



## Order from the pheatmap dendrogram
marker_order <- rownames(cancer_corr_MTAP_status[c(4:10,12:19,21:92),1:7][heatmap_order$tree_row$order,])

#To pick out the corr values in cancer_corr_MTAP_status where the p value in cancer_corr_Pval_MTAP_status < 0.05)
cancer_corr_sig <- ifelse(cancer_corr_Pval_MTAP_status %>% as.matrix < 0.05, cancer_corr_MTAP_status %>% as.matrix, NA)
str(cancer_corr_sig)

# plot filter for significany correlations
pheatmap(cancer_corr_sig[marker_order,1:7], cluster_rows = F, cluster_cols = F )




```

CD200 cor CD200R1
```{r fig.height=2, fig.width=12}
cancer_corr_CD200R1_CD200 %>% t %>% data.frame %>% rownames_to_column("Cancer") %>% rename(Correlation="CD200") %>% mutate(Cancer=fct_reorder(Cancer,-Correlation)) %>%
  ggplot(aes(x=Cancer,y=Correlation,fill=Correlation)) +
  geom_chicklet(stat="identity") +
  theme_bw() +
  ylab("CD200-CD200R1 expression correlation") +
  geom_hline(yintercept = 0) +
  scale_fill_gradient2(low="dodgerblue3",mid="white",high="red")
```

Plot to match Correlation
Plot cases that are Dual marker positive > 10 TPM
```{r fig.height=5, fig.width=14}
get order of cancers 
my_order <- cancer_corr_CD200R1_CD200 %>% t %>% data.frame %>% rownames_to_column("Cancer") %>% rename(Correlation="CD200") %>% mutate(Cancer=fct_reorder(Cancer,-Correlation)) %>% arrange(-Correlation) %>% pull(Cancer) 
my_order <- as.vector(my_order)

SLC_TCGA_meta %>% filter(TCGA_Study %in% c("PAAD","BRCA","PRAD","LUAD","THYM","KIRC","SARC","SKCM","READ","STAD","COAD","LUSC","THCA","BLCA","UCEC","HNSC","PCPG","LIHC","ESCA","CESC","LGG","KIRP","GBM","OV","LAML","TGCT") ) %>% mutate(TCGA_Study=factor(TCGA_Study,levels=c("PAAD","BRCA","PRAD","LUAD","THYM","KIRC","SARC","SKCM","READ","STAD","COAD","LUSC","THCA","BLCA","UCEC","HNSC","PCPG","LIHC","ESCA","CESC","LGG","KIRP","GBM","OV","LAML","TGCT") )) %>% select(TCGA_Study,CD200,CD200R1) %>% gather(Marker,Log2_TPM,-TCGA_Study) %>%
  ggplot(aes(x=TCGA_Study, y=Log2_TPM,color=TCGA_Study)) +
  geom_quasirandom(show.legend = T, size=0.5) +
  theme_bw() +
  facet_wrap(~Marker,ncol=1, scales="free_y") +
  stat_summary(fun="median", geom="crossbar",color="grey50") +
  #stat_n_text(text.box=F) +
  theme(axis.text.x=element_text(angle=35,size=11,hjust=1),strip.text=element_text(size=11)) +
  ylab("Log2(TPM)")

#Plot percent dual marker positive >=10 TPM: they're all positive

SLC_TCGA_meta %>% filter(TCGA_Study %in% c("PAAD","BRCA","PRAD","LUAD","THYM","KIRC","SARC","SKCM","READ","STAD","COAD","LUSC","THCA","BLCA","UCEC","HNSC","PCPG","LIHC","ESCA","CESC","LGG","KIRP","GBM","OV","LAML","TGCT") ) %>% mutate(TCGA_Study=factor(TCGA_Study,levels=c("PAAD","BRCA","PRAD","LUAD","THYM","KIRC","SARC","SKCM","READ","STAD","COAD","LUSC","THCA","BLCA","UCEC","HNSC","PCPG","LIHC","ESCA","CESC","LGG","KIRP","GBM","OV","LAML","TGCT") )) %>% mutate(Double_positive=if_else(CD200>=3.32 & CD200R1>=3.32,1,0)) %>% group_by(TCGA_Study) %>% summarise(Pct_Double_Positive=sum(Double_positive==1)*100/n())
```
Find the TH2 signal across Tumors
```{r fig.height=4, fig.width=14}
SLC_TCGA_meta %>% ungroup %>% select(TCGA_Study,Th2_Cells) %>% mutate(TCGA_Study= fct_reorder(TCGA_Study,-Th2_Cells)) %>%
  ggplot(aes(x=TCGA_Study, y=Th2_Cells,color=TCGA_Study)) +
  geom_quasirandom(dodge.width = 0.8, show.legend = T, size=0.5) +
  theme_bw() +
  stat_summary(fun="median", geom="crossbar",color="grey50") +
  #stat_n_text(text.box=F) +
  theme(axis.text.x=element_text(angle=35,size=11,hjust=1),strip.text=element_text(size=11)) +
  ylab("Predicted Th2 cell content (arbitrary units)")
```


There is negative correlation between B3GNT2/CD3E with IFN-Gamma response
```{r fig.height=5, fig.width=5}

#Normalized to CD3E content
SLC_TCGA_meta %>% mutate(B3GNT2_norm_bin=ntile(B3GNT2_norm_CD3E,2),B3GNT2_norm_bin=if_else(B3GNT2_norm_bin==1,"Low","High")) %>% filter(TCGA_Study=="LUAD") %>%
  ggplot(aes(x=B3GNT2_norm_bin,y=`IFN-gamma_Response`)) +
  geom_boxplot() +
  geom_quasirandom() +
  theme_bw()  +
  stat_compare_means(method="t.test")

SLC_TCGA_meta %>% mutate(B3GNT2_norm_bin=ntile(B3GNT2_norm_CD3E,2),B3GNT2_norm_bin=if_else(B3GNT2_norm_bin==1,"Low","High")) %>% filter(TCGA_Study=="LUAD") %>%
ggplot(aes(x=B3GNT2_norm_bin,y=B3GNT2_norm_CD3E)) +
  geom_boxplot() +
  geom_quasirandom() +
  theme_bw()  +
  stat_compare_means(method="t.test")


#Not normalized to CD3E content:
SLC_TCGA_meta %>% mutate(B3GNT2_norm_bin=ntile(B3GNT2_norm_CD3E,2),B3GNT2_norm_bin=if_else(B3GNT2_norm_bin==1,"Low","High")) %>% mutate(B3GNT2_bin=ntile(B3GNT2,2),B3GNT2_bin=if_else(B3GNT2_bin==1,"Low","High")) %>% filter(TCGA_Study=="LUAD") %>%
  ggplot(aes(x=B3GNT2_bin,y=`IFN-gamma_Response`)) +
  geom_boxplot() +
  geom_quasirandom() +
  theme_bw() +
  stat_compare_means(method="t.test")

SLC_TCGA_meta %>% mutate(B3GNT2_norm_bin=ntile(B3GNT2_norm_CD3E,2),B3GNT2_norm_bin=if_else(B3GNT2_norm_bin==1,"Low","High")) %>% mutate(B3GNT2_bin=ntile(B3GNT2,2),B3GNT2_bin=if_else(B3GNT2_bin==1,"Low","High")) %>% filter(TCGA_Study=="LUAD") %>%
ggplot(aes(x=B3GNT2_bin,y=B3GNT2)) +
  geom_boxplot() +
  geom_quasirandom() +
  theme_bw()  +
  stat_compare_means(method="t.test")

#Plot B3GNT2 when Binned for CD3E
SLC_TCGA_meta %>% filter(TCGA_Study=="LUAD") %>% mutate(CD3E_bin=ntile(CD3E,2),CD3E_bin=if_else(CD3E_bin==1,"Low","High")) %>%
  ggplot(aes(x=CD3E_bin,y=B3GNT2)) +
  geom_boxplot() +
  geom_quasirandom() +
  theme_bw()  +
  stat_compare_means(method="t.test")
```



TCGA KM Plots
OS KM analysis

```{r fig.height=9, fig.width=12}
my_marker <- c("PDCD1","CXCR3","CD8A","CD3E","TCF7")
my_marker <- c("SPPL3")
my_marker <- c("SIGLEC10","CLEC4E","STAB1","HCK","SIGLEC15")
  
load(file="/fcrbiouatappn01/home/mxu4/TCGA/EBPlusPlusAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp_vst_df_MX.rdata") 

SLC_TCGA <- RNAseq_vst_df %>% filter(Gene %in% c(my_marker,"CD3E")) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample.ID") %>% mutate(Sample.ID= str_replace_all(Sample.ID,"\\.","-")) %>% mutate(Sample.ID = substr(Sample.ID,1,16)) %>% mutate(Sample_type= substr(Sample.ID,14,16)) %>% select(Sample.ID,Sample_type,everything()) %>% mutate(Tissue=if_else(grepl("11",Sample_type),"Normal","Tumor")) %>% mutate(Tissue=factor(Tissue,levels=c("Tumor","Normal"))) %>% select(-Sample_type) %>% mutate(Sample.ID = substr(Sample.ID,1,12)) %>% filter(Tissue=="Tumor")

#TCGA Metadata
load(file="/fcrbiouatappn01/home/mxu4/TCGA/TCGA_ImmuneLandscapeOfCancer_Tumors_ImmuneScores.rdata") # TCGA_TMB_IOscore

TCGA_TMB_IOscore

#Merge meta data with mRNA expression data for plotting

SLC_TCGA_meta <- SLC_TCGA %>% left_join(TCGA_TMB_IOscore %>% select(TCGA_Participant_Barcode,TCGA_Study,TCGA_Subtype,OS_Time,OS), by=c("Sample.ID"="TCGA_Participant_Barcode")) %>% filter(!is.na(TCGA_Study))

#Single marker
SLC_TCGA_meta %<>% mutate(CLEC4E_bin=ntile(CLEC4E,2),HCK_bin=ntile(HCK,2),SIGLEC10_bin=ntile(SIGLEC10,2),SIGLEC15_bin=ntile(SIGLEC15,2),STAB1_bin=ntile(STAB1,2))
#SLC_TCGA_meta %<>% mutate(CD5_norm=CD5/CD3E,UBASH3A_norm=UBASH3A/CD3E,UBASH3B_norm=UBASH3B/CD3E) %>% mutate(CD5_bin=ntile(CD5_norm,2),UBASH3A_bin=ntile(UBASH3A_norm,2),UBASH3B_bin=ntile(UBASH3B_norm,2))

#Normalize to Myeloid content: Determine High low expression by each indication
SLC_TCGA_meta_myeloid <- SLC_TCGA_meta %>% filter(Tissue=="Tumor") %>% select(-Tissue,-TCGA_Subtype,-Sample.ID) %>% gather(Marker,mRNA_expression,-TCGA_Study,-OS_Time,-OS,-Macrophages) %>% mutate(mRNA_norm=mRNA_expression/Macrophages) %>% group_by(TCGA_Study,Marker) %>% mutate(Marker_norm_Bin=ntile(mRNA_norm,2),Marker_Bin=ntile(mRNA_expression,2)) %>% mutate(Marker_norm_Bin=if_else(Marker_norm_Bin==1,"Low_bin","High_Bin"),Marker_Bin=if_else(Marker_Bin==1,"Low_bin","High_Bin")) %>% ungroup() %>% select(-Macrophages,-mRNA_expression,-mRNA_norm) %>% filter(!is.na(Marker_Bin))

#Normalize to CD68 content: Determine High low expression by each indication
SLC_TCGA_meta_CD68 <- SLC_TCGA_meta %>% filter(Tissue=="Tumor") %>% select(-Tissue,-TCGA_Subtype,-Sample.ID) %>% gather(Marker,mRNA_expression,-TCGA_Study,-OS_Time,-OS,-Macrophages,-CD68) %>% mutate(mRNA_norm=mRNA_expression/CD68) %>% group_by(TCGA_Study,Marker) %>% mutate(Marker_norm_Bin=ntile(mRNA_norm,2),Marker_Bin=ntile(mRNA_expression,2)) %>% mutate(Marker_norm_Bin=if_else(Marker_norm_Bin==1,"Low_bin","High_Bin"),Marker_Bin=if_else(Marker_Bin==1,"Low_bin","High_Bin")) %>% ungroup() %>% select(-Macrophages,-mRNA_expression,-mRNA_norm) %>% filter(!is.na(Marker_Bin))

#Normalize to CD3 content: Determine High low expression by each indication
SLC_TCGA_meta_CD3 <- SLC_TCGA_meta %>% filter(Tissue=="Tumor") %>% select(-Tissue,-TCGA_Subtype,-Sample.ID) %>% gather(Marker,mRNA_expression,-TCGA_Study,-OS_Time,-OS,-CD3E) %>% mutate(mRNA_norm=mRNA_expression/CD3E) %>% group_by(TCGA_Study,Marker) %>% mutate(Marker_norm_Bin=ntile(mRNA_norm,2),Marker_Bin=ntile(mRNA_expression,2)) %>% mutate(Marker_norm_Bin=if_else(Marker_norm_Bin==1,"Low_bin","High_Bin"),Marker_Bin=if_else(Marker_Bin==1,"Low_bin","High_Bin")) %>% ungroup() %>% select(-mRNA_expression,-mRNA_norm) %>% filter(!is.na(Marker_Bin))

# Markers
SLC_TCGA_meta_km <- SLC_TCGA_meta %>% mutate(CXCR3_norm = CXCR3/CD3E,PD1_norm = PDCD1/CD3E,TCF7_norm = TCF7/CD3E) %>% select(-Tissue) %>% group_by(TCGA_Study) %>% mutate(TCF7_bin=ntile(TCF7,2),TCF7_norm_bin=ntile(TCF7_norm,2),CXCR3_norm_bin=ntile(CXCR3_norm,2),PD1_norm_bin=ntile(PD1_norm,2),CXCR3_PD1_high_bin=if_else(CXCR3_norm_bin==2 & PD1_norm_bin==2,2,1)) %>% ungroup() %>% mutate(TCF7_bin=if_else(TCF7_bin==1,"Low","High"),TCF7_norm_bin=if_else(TCF7_norm_bin==1,"Low","High"),CXCR3_norm_bin=if_else(CXCR3_norm_bin==1,"Low","High"),PD1_norm_bin=if_else(PD1_norm_bin==1,"Low","High"),CXCR3_PD1_high_bin=if_else(CXCR3_PD1_high_bin==1,"Low","High")) %>% filter(!is.na(OS_Time),!is.na(CXCR3_PD1_high_bin),!is.na(TCGA_Study)) %>% select(-2,-3,-4,-5,-6,-8,-11)


require("survival")
library("survminer")
#fit2 <- survfit(Surv(OS_Time,OS) ~ CXCR3_PD1_high_bin+TCGA_Study, data=SLC_TCGA_meta_km)

ggsurv <- ggsurvplot(fit2, conf.int = TRUE)
ggsurv$plot+facet_wrap(~TCGA_Study)

# to plot by facet, remove all optional plots
ggsurvplot(
    #fit = survfit(Surv(OS_Time,OS) ~ CXCR3_PD1_high_bin, data=SLC_TCGA_meta_km),
    fit = survfit(Surv(OS_Time,OS) ~ Marker_norm_Bin, data=SLC_TCGA_meta_CD3),
    #fit = survfit(Surv(OS_Time,OS) ~ CLEC4E_bin, data=SLC_TCGA_meta),
    #fit = survfit(Surv(OS_Time,OS) ~ SIGLEC10_bin, data=SLC_TCGA_meta),
    #fit = survfit(Surv(OS_Time,OS) ~ Marker_Bin, data=SLC_TCGA_meta_myeloid %>% filter(Marker=="SIGLEC10")),
    #fit = survfit(Surv(OS_Time,OS) ~ STAB1_bin, data=SLC_TCGA_meta),
    #xlab = "Days", 
    #ylab = "Overall survival probability",
    #risk.table=TRUE, 
    #size=2,
    #censor=OS,
    xlim=c(0,3000),
    pval=TRUE,
    facet.by="TCGA_Study") 

ggsurvplot(
    fit = survfit(Surv(OS_Time,OS) ~ Marker_norm_Bin, data=SLC_TCGA_meta_CD3 %>% filter(Marker=="UBASH3A")),
    xlim=c(0,3000),
    #pval=TRUE,
    facet.by="TCGA_Study") 

 
survplot$plot + facet_wrap(~ TCGA_Study) 

c("SIGLEC10","CLEC4E","STAB1","HCK","SIGLEC15")
```
```{r}
ggsurvplot(
    fit = survfit(Surv(OS_Time,OS) ~ Marker_Bin, data=SLC_TCGA_meta_CD68 %>% filter(Marker=="STAB1",TCGA_Study=="OV")),
    xlim=c(0,4000),
    pval=TRUE,
    facet.by="TCGA_Study") 
```



PanTCGA OS

Stats
```{r}
ggsurvplot(
    #fit = survfit(Surv(OS_Time,OS) ~ CXCR3_PD1_high_bin, data=SLC_TCGA_meta_km),
    #fit = survfit(Surv(OS_Time,OS) ~ TCF7_norm_bin, data=SLC_TCGA_meta_km),
    fit = survfit(Surv(OS_Time,OS) ~ SPPL3_Bottom20, data=SPPL3_TCGA_meta_km),
    pval=TRUE,
    xlim=c(0,5000))
```

Calculate P values for all 

```{r}
c("SIGLEC10","CLEC4E","STAB1","HCK","SIGLEC15")
#
fit = survfit(Surv(OS_Time,OS) ~ Marker_Bin, data=SLC_TCGA_meta_myeloid %>% filter(Marker=="CLEC4E"))

#
diff<- survdiff( Surv(OS_Time,OS)~ Marker_Bin, data=SLC_TCGA_meta_myeloid  %>% filter(Marker=="CLEC4E",TCGA_Study=="SKCM")) 
pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE)

# If the first number is higher than the second, then more deaths and worse survival, if first number is lower than the second, then better survival
diff$obs[1] < diff$obs[2]

my_indications <- SLC_TCGA_meta_CD3 %>% filter(!is.na(TCGA_Study)) %>% select(TCGA_Study) %>% unique %>% pull(TCGA_Study)
length(my_indications)
#function to return the p value and survival effect polarity
pvals <- lapply(my_indications,function(i) {
  diff <-survdiff( Surv(OS_Time,OS) ~ Marker_norm_Bin, data= SLC_TCGA_meta_CD3 %>% filter(Marker=="UBASH3A",TCGA_Study==i))
   #fit <-survfit( Surv(OS_Time,OS) ~ Marker_Bin, data= SLC_TCGA_meta_myeloid %>% filter(Marker=="HCK",TCGA_Study==i))
  HR <- (diff$obs[1]/diff$exp[1])/(diff$obs[2]/diff$exp[2])
  Surv_effect <- if_else(diff$obs[1] < diff$obs[2],"Positive","Negative")
  return(list(pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE),Surv_effect,HR )  )
})
my_pvals <- do.call(rbind,pvals) %>% unlist %>% matrix(nrow=33,ncol=3) %>% as.data.frame()
rownames(my_pvals) <- my_indications
colnames(my_pvals) <- c("P_val","Effect","HR")

#normalized to Mac
CLEC4E_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study")
SIGLEC10_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study")
STAB1_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study")
HCK_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study")
SIGLEC15_surv <-  my_pvals %>% tibble::rownames_to_column("TCGA_Study")

# STAB1 normalized to CD68
STAB1_CD68_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study")

#Not normalized
CLEC4E_orig_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study") #
SIGLEC10_orig_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study") #
STAB1_orig_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study") #
HCK_orig_surv <- my_pvals %>% tibble::rownames_to_column("TCGA_Study") #
SIGLEC15_orig_surv <-  my_pvals %>% tibble::rownames_to_column("TCGA_Study") #

CD5_surv <-  my_pvals %>% tibble::rownames_to_column("TCGA_Study")
UBASH3B_surv <-  my_pvals %>% tibble::rownames_to_column("TCGA_Study")
UBASH3A_surv <-  my_pvals %>% tibble::rownames_to_column("TCGA_Study")

KM_markers <- CLEC4E_surv %>% left_join(SIGLEC10_surv, by="TCGA_Study") %>% left_join(STAB1_surv, by="TCGA_Study") %>% left_join(HCK_surv, by="TCGA_Study") %>% left_join(SIGLEC15_surv, by="TCGA_Study") %>% gather(Marker,P_value,-TCGA_Study) %>% mutate(Neg_log10=-log(P_value,10)) %>% mutate(Effect=2)
```
# Plot HR

```{r fig.height=5, fig.width=4.5}
CLEC4E_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

CLEC4E_orig_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

STAB1_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

STAB1_orig_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

HCK_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

HCK_orig_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

SIGLEC10_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

SIGLEC10_orig_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

STAB1_CD68_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

CD5_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

UBASH3B_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()

UBASH3A_surv %>% mutate(HR=as.numeric(HR)) %>%
  ggplot(aes(x=TCGA_Study,y=HR, color=P_val <= 0.05)) +
  theme_bw() +
  coord_flip() +
  geom_point(size=2) +
  geom_hline(yintercept=1, linetype="dashed") +
  scale_colour_manual(values=c("grey40","red")) +
  scale_y_log10()
```



# Plot a Heatmap
```{r fig.height=8, fig.width=5.5}
KM_markers %>%
  ggplot(aes(y=TCGA_Study,x=Marker,colour=Effect)) +
  geom_point(size=0.1,colour="grey40") +
  theme_bw() +
  geom_point(data=subset(KM_markers, P_value <= 0.05),aes(size=Neg_log10), colour="red") +
  scale_color_manual(values=c("blue","red")) +
  scale_x_discrete(limits=rev) +
  scale_y_discrete(limits=rev) +
  xlab("Marker") +
  theme(axis.text.x=element_text(angle=0,size=9),strip.text=element_text(size=10))
```

#######################
###################
###############

Load POPLAR OAK data

###############
####################
########################

```{r}

POPLAR_OAK_meta <- read.table(file="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/IO_Cohorts/OAK_POPLAR_NSCLC_Atezo/OAK_POPLAR_PatientInfo.csv", sep=",",header=T) %>% dplyr::rename(SampleID="X")

load(file="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/IO_Cohorts/OAK_POPLAR_NSCLC_Atezo/POPLAR_OAK_log2TPM.rdata")
SLC7A11_mRNA <- POPLAR_OAK_mRNA %>% filter(Gene %in% c("SIGLEC10","CLEC4E","STAB1","HCK","SIGLEC15","CD68")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("SampleID")

SLC7A11_mRNA <- POPLAR_OAK_mRNA %>% filter(Gene %in% c(my_marker,"CD3E")) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("SampleID")
#SLC7A11_mRNA <- POPLAR_OAK_mRNA %>% filter(Gene %in% c("CXCR3","PDCD1","TCF7","CD3E")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("SampleID")

SLC7A11_mRNA %<>% gather(Marker,Expression,-SampleID,-CD3E) %>% group_by(Marker) %>% mutate(Marker_Bin=ntile(Expression,2),Expression_norm=Expression/CD3E,Marker_norm_Bin=ntile(Expression_norm,2)) %>% mutate(Marker_norm_Bin=if_else(Marker_norm_Bin==1,"Low_bin","High_Bin"))

#normalizing to CD68 doesn't work because there are a lot of 0 values
```

KM OS analysis
```{r fig.height=5, fig.width=7}
library(survminer)

#combine RNA with meta
POPLAR_OAK_data <- POPLAR_OAK_meta %>% left_join(SLC7A11_mRNA,by="SampleID")

  ggsurvplot(
    fit = survfit(Surv(OS_MONTHS,OS_CENSOR) ~ Marker_norm_Bin, data=POPLAR_OAK_data %>% filter(Marker=="CD5",ACTARM!="yg")),
    xlab = "Months", 
    ylab = "Overall survival probability",
    pval=TRUE,
    fontsize=4,
    facet.by=c("HIST","ACTARM")) + theme_bw()

```




Plot CPTAC: tumor mRNA and Mass Spec data

Load data
```{r}
load("/fcrbiouatappn01/resbioinfo/data/Oncology/Target_Evaluation_Project/Datasets/CPTAC/CPTAC_mRNA_batch_Quantile_norm_log2TPM.rdata") # save(CPTAC_meta,file="/fcrbiouatappn01/home/mxu4/reference/CPTAC/CPTAC_meta.rdata") # 1463 2 
load("/fcrbiouatappn01/resbioinfo/data/Oncology/Target_Evaluation_Project/Datasets/CPTAC/CPTAC_meta.rdata") 
load("/fcrbiouatappn01/resbioinfo/data/Oncology/Target_Evaluation_Project/Datasets/CPTAC/CPTAC_protein.rdata") #  CPTAC_p
```

Plot SLC7A11 from the normalized data

```{r fig.height=5, fig.width=12}
My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% rownames_to_column("Gene") %>% filter(Gene %in% c("SLC7A11")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("Sample") %>% filter(SLC7A11 >= -3)

My_p <- CPTAC_p %>% filter(Gene=="SLC7A11") %>% column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(SLC7A11_p=as.numeric(SLC7A11)) %>% select(-SLC7A11) %>% rownames_to_column("Sample")

My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% tibble::rownames_to_column("Gene") %>% filter(Gene %in% c("CD5")) %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% tibble::rownames_to_column("Sample") %>% filter(CD5 >= -3)

My_p <- CPTAC_p %>% filter(Gene=="CD5") %>% tibble::column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(CD5_p=as.numeric(CD5)) %>% select(-CD5) %>% tibble::rownames_to_column("Sample")


#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(SLC7A11_p)) %>%
  ggplot(aes(x=SLC7A11,y=SLC7A11_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=4,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("SLC7A11 (log2TPM)") +
  ylab("SLC7A11 norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))

#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(CD5_p)) %>%
  ggplot(aes(x=CD5,y=CD5_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=4,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("CD5 (log2TPM)") +
  ylab("CD5 norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))
```

Plot CD68 from the normalized data

```{r fig.height=6, fig.width=11}
My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% rownames_to_column("Gene") %>% filter(Gene %in% c("CD68")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("Sample") %>% filter(CD68 >= -3)

My_p <- CPTAC_p %>% filter(Gene=="CD68") %>% column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(CD68_p=as.numeric(CD68)) %>% select(-CD68) %>% rownames_to_column("Sample")

#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(CD68_p)) %>%
  ggplot(aes(x=CD68,y=CD68_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=3,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("CD68 (log2TPM)") +
  ylab("CD68 norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))
```

Plot CD163 from the normalized data

```{r fig.height=8, fig.width=11}
My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% rownames_to_column("Gene") %>% filter(Gene %in% c("CD163")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("Sample") %>% filter(CD163 >= -3)

My_p <- CPTAC_p %>% filter(Gene=="CD163") %>% column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(CD163_p=as.numeric(CD163)) %>% select(-CD163) %>% rownames_to_column("Sample")

#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(CD163_p)) %>%
  ggplot(aes(x=CD163,y=CD163_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=3,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("CD163 (log2TPM)") +
  ylab("CD163 norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))
```
Plot IL18RAP from the normalized data

```{r fig.height=8, fig.width=11}
My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% rownames_to_column("Gene") %>% filter(Gene %in% c("IL18RAP")) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("Sample") %>% filter(IL18RAP >= -3)

My_p <- CPTAC_p %>% filter(Gene=="IL18RAP") %>% column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(IL18RAP_p=as.numeric(IL18RAP)) %>% select(-IL18RAP) %>% rownames_to_column("Sample")

#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(CD163_p)) %>%
  ggplot(aes(x=IL18RAP,y=IL18RAP_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=3,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("IL18RAP (log2TPM)") +
  ylab("IL18RAP norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))
```

Plot ImmProt: Protein Mass Spec data from healthy PBMC sorted cells 

```{r fig.height=3, fig.width=9}
my_marker <- "SPPL3"
My_mRNA <- CPTAC_TPM_batch_norm_log2 %>% rownames_to_column("Gene") %>% filter(Gene %in% c(my_marker)) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% rownames_to_column("Sample") %>% filter(my_marker >= -3)

My_p <- CPTAC_p %>% filter(Gene==my_marker) %>% column_to_rownames("Gene") %>% t %>% data.frame %>% mutate(SPPL3_p=as.numeric(my_marker)) %>% select(-my_marker) %>% rownames_to_column("Sample")

#filter out the mRNA = -6 samples, where the mRNA signal is actually 0
My_mRNA %>% left_join(My_p,by="Sample") %>% left_join(CPTAC_meta,by=c("Sample"="Patient_ID")) %>% filter(!is.na(SLC7A11_p)) %>%
  ggplot(aes(x=SLC7A11,y=SLC7A11_p)) +
  geom_point(aes(colour=Tissue_type)) +
  theme_classic() +
  facet_wrap(~Cancer, ncol=4,scales="free") + 
  geom_smooth(method="lm") +
  stat_cor(method="pearson") +
  xlab("SLC7A11 (log2TPM)") +
  ylab("SLC7A11 norm Mass Spec abundance") +
  scale_color_manual(values=c("grey30","red"))
```



Load ImmProt data

```{r}
load("/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/ImmProt/ImmProt_Protein_CopyNum_Log10_mean.rdata")
```

Data processing: no need to run
```{r}
ImmProt_Protein_CopyNum <- readxl::read_xlsx(path="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/ImmProt/ImmProt_CopyNumber_Supp.xlsx",sheet="20160822_tableS4_protein.copy.n",col_names=T) %>% select(-`Majority protein IDs`) %>% separate(`Gene names`,into=c("Gene",NA),remove=T) %>% group_by(Gene) %>% summarise_all(sum) %>% ungroup %>% unique %>% filter(!is.na(Gene), !grepl("^[[:digit:]]+$",Gene)) %>% column_to_rownames("Gene") %>% as.matrix

ImmProt_Protein_CopyNum_log10 <- log10(ImmProt_Protein_CopyNum+1)

save(ImmProt_Protein_CopyNum_log10,file="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/ImmProt/ImmProt_Protein_CopyNum_Log10.rdata")


ImmProt_Protein_CopyNum_log10_mean <- ImmProt_Protein_CopyNum_log10 %>% t %>% data.frame %>% rownames_to_column("Sample") %>% separate(Sample,into=c(NA,"Celltype",NA,"Cell_state"),sep="_") %>% mutate(Sample=paste0(Celltype,"_",Cell_state)) %>% select(-Celltype,-Cell_state) %>% group_by(Sample) %>% summarise_all(mean) %>% ungroup %>% unique %>% column_to_rownames("Sample") %>% t

save(ImmProt_Protein_CopyNum_log10_mean,file="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/ImmProt/ImmProt_Protein_CopyNum_Log10_mean.rdata")

ImmProt_Protein_CopyNum_log10_mean %>% data.frame %>% rownames_to_column("Gene") %>% write.table(file="/fcrbiouatappn01/resbioinfo/data/Oncology/Team_Shared/ImmProt/ImmProt_Protein_CopyNum_Log10_mean.txt",sep="\t",quote=F,row.names = F)
```

Pull out list of proteins for Raj's T cell Signal II
Missing:
HAVCR1
TNFRSF14/HVEM
TNFRSF25/DR3
```{r}
Raj_ImmProt <- ImmProt_Protein_CopyNum_log10_mean %>% data.frame %>% rownames_to_column("Gene") %>% filter(Gene %in% c("CD28","ICOS","CD226","CD2","SLAMF1","CRTAM","HAVCR1","TNFRSF9","CD27","TNFRSF4","TNFRSF14","TNFSF14","CD40LG","TNFRSF25","TNFRSF18","TNFRSF8")) %>% mutate(Gene=factor(Gene,levels=c("CD28","ICOS","CD226","CD2","SLAMF1","CRTAM","HAVCR1","TNFRSF9","CD27","TNFRSF4","TNFRSF14","TNFSF14","CD40LG","TNFRSF25","TNFRSF18","TNFRSF8"))) %>% arrange(Gene) %>% column_to_rownames("Gene") %>% as.matrix


SIRPa_ImmProt <- ImmProt_Protein_CopyNum_log10_mean %>% data.frame %>% rownames_to_column("Gene") %>% filter(Gene %in% c("SIRPA","CD47")) %>% mutate(Gene=factor(Gene,levels=c("SIRPA","CD47"))) %>% arrange(Gene) %>% column_to_rownames("Gene") %>% as.matrix

```

Plot data as Heatmap
```{r fig.height=7, fig.width=12}
# Raj's data
paletteLength <- 50
myColor <- colorRampPalette(c("white", "red"))(paletteLength)
myBreaks <- c(seq(min(Raj_ImmProt,rm.na=T), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(Raj_ImmProt,rm.na=T)/paletteLength, max(Raj_ImmProt,rm.na=T), length.out=floor(paletteLength/2)))

pheatmap(Raj_ImmProt, cluster_cols = F, cluster_rows=F,cellwidth=15,cellheight=15,color=myColor) # , breaks=myBreaks, color=myColor

# Sirpa data
paletteLength <- 50
myColor <- colorRampPalette(c("white", "red"))(paletteLength)
myBreaks <- c(seq(min(SIRPa_ImmProt,rm.na=T), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(SIRPa_ImmProt,rm.na=T)/paletteLength, max(SIRPa_ImmProt,rm.na=T), length.out=floor(paletteLength/2)))

pheatmap(SIRPa_ImmProt, cluster_cols = F, cluster_rows=F,cellwidth=15,cellheight=15,color=myColor) # , breaks=myBreaks, color=myColor
```
Plot Bar Graph for a single marker
```{r fig.height=4, fig.width=9}
ImmProt_Protein_CopyNum_log10_mean %>% data.frame %>% rownames_to_column("Gene") %>% filter(Gene %in% c("SIRPA")) %>% gather(Cell_type,Log10_ProteinCopyNumber,-Gene) %>% separate(Cell_type,into=c("Cell_type","State"),sep="_") %>%
  ggplot(aes(x=Cell_type,y=Log10_ProteinCopyNumber,fill=Cell_type)) +
  geom_chicklet(show.legend = F) +
  facet_wrap(~State,ncol=1) +
  theme_bw() +
  theme(plot.margin = unit(c(0.5,0.5,0.5,2), "cm")) +
  theme(axis.text.x=element_text(angle=35,size=11,hjust=1))

```

#########################
##################

TISMO

##################
###########################

Load data
```{r}
# load meta data
vitro_meta <- read.table(file="/fcrbiouatappn01/home/mxu4/reference/TISMO/TISMO_vitrosample_annotations.csv", header=T, sep=",")
vivo_meta <- read.table(file="/fcrbiouatappn01/home/mxu4/reference/TISMO/TISMO_vivosample_annotations.csv", header=T, sep=",")

# cell line meta
celline_meta <- read.table(file="/fcrbiouatappn01/home/mxu4/reference/TISMO/TISMO_cellline_annotations.text", header=T, sep="\t")

# load RNA seq data
tismo_vitro <- readRDS(file="/fcrbiouatappn01/home/mxu4/reference/TISMO/TISMO_expressionvitro_profiles.RDS")
tismo_vivo <- readRDS(file="/fcrbiouatappn01/home/mxu4/reference/TISMO/TISMO_expressionvivo_profiles.RDS")
```

```{r}
vitro_meta
```

mRNA Slc7a11

```{r}
vitro <- vitro_meta %>% select(Cell_Line,Baseline,Cell_genotype,SampleName,Cancer_type) %>% mutate(Type="In_Vitro")
vivo <- vivo_meta %>% select(Cell_Line,ICB,Cell_genotype,SampleName,Cancer_type) %>% rename(Baseline="ICB") %>% mutate(Type="In_Vivo")
vitro_vivo <- rbind(vitro,vivo)

vitro_vivo_slc7a11 <- rbind(tismo_vitro["Slc7a11",] %>% t,tismo_vivo["Slc7a11",] %>% t) %>% data.frame %>% tibble::rownames_to_column("SampleName") 
vitro_vivo_slc7a11 %<>% left_join(vitro_vivo,by="SampleName")
```

Plot mRNA Slc7a11
```{r fig.height=16, fig.width=8}
vitro_vivo_slc7a11 %>% mutate(Cell_Line=fct_reorder(Cell_Line,Slc7a11)) %>% filter(!is.na(Cell_Line)) %>%
  ggplot(aes(x=Cell_Line,y=Slc7a11, colour=Type)) +
  geom_boxplot(position="dodge", dodge_width=0.8) +
  geom_point(position="dodge", dodge_width=0.9) +
  coord_flip() +
  scale_colour_manual(values=c("grey40","red")) +
  theme_bw()
```
 List of syngeneic models
```{r}
vitro_vivo_slc7a11 %>% mutate(Cell_Line=fct_reorder(Cell_Line,Slc7a11)) %>% filter(!is.na(Cell_Line)) %>% select(Cell_Line,Cancer_type) %>% tibble::remove_rownames() %>% arrange(Cell_Line) %>% unique

```
 
Nrf2 Signature

# this is not finished being re-coded yet
```{r}
Singh_NRF2_sig <- read.table(file="/fcrbiouatappn01/home/mxu4/New_Targets_Jackson_Group/SLC7A11/Singh_NRF2_Signature.txt",header=T) %>% pull(Singh_NRF2_Sig)
load("/fcrbiouatappn01/home/mxu4/reference/Rat_Human_Mouse_HUGO_homolog_table.rdata") #Rat_Human_Mouse
Singh_NRF2_sig_mm <- Rat_Human_Mouse %>% filter(Human %in% Singh_NRF2_sig) %>% pull(Mouse)

tismo_vitro[Singh_NRF2_sig_mm,]
tismo_vivo[Singh_NRF2_sig_mm,]
vitro_vivo_Sig <- cbind(tismo_vitro[Singh_NRF2_sig_mm,],tismo_vivo[Singh_NRF2_sig_mm,])

mx_zscore <- function(i) {
  require(dplyr)
  denom <- ifelse(sd(i,na.rm=T) >= 0.1,sd(i,na.rm=T),0.1)
  result <- (i-mean(i,na.rm=T)) / denom
}
vitro_vivo_Sig_z <- t(apply(vitro_vivo_Sig,1,mx_zscore))
vitro_vivo_Sig_score <- vitro_vivo_Sig_z %>% data.frame %>% summarise_all(mean)

vitro_vivo_nrf2_sig <- data.frame(Nrf2_Sig_Score=vitro_vivo_Sig_score %>% t) %>% tibble::rownames_to_column("SampleName") %>% left_join(vitro_vivo,by="SampleName")
```
Plot NRF2 Signature
```{r fig.height=16, fig.width=8}
vitro_vivo_nrf2_sig %>% mutate(Cell_Line=fct_reorder(Cell_Line,Nrf2_Sig_Score)) %>% filter(!is.na(Cell_Line)) %>%
  ggplot(aes(x=Cell_Line,y=Nrf2_Sig_Score, colour=Type)) +
  geom_boxplot(position="dodge", width=0.8) +
  geom_point(position="dodge", width=0.9) +
  coord_flip() +
  scale_colour_manual(values=c("grey40","red")) +
  theme_bw()
```
 List of syngeneic models
```{r}
vitro_vivo_nrf2_sig %>% mutate(Cell_Line=fct_reorder(Cell_Line,Nrf2_Sig_Score)) %>% filter(!is.na(Cell_Line)) %>% select(Cell_Line,Cancer_type) %>% tibble::remove_rownames() %>% arrange(Cell_Line) %>% unique %>% rev()

```

Get annotations from BioMart
```{r}
library(biomaRt)
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("biomaRt")

listEnsembl()
ensembl <- useEnsembl(biomart = "ensembl",dataset="mfascicularis_gene_ensembl")
searchAttributes(mart = ensembl, pattern = "hgnc")
MF_genes_ENSEMBL_HGNC <- getBM(attributes=c("hgnc_symbol","hgnc_id","ensembl_gene_id"),
      mart=ensembl) # 
# the number of genes with hgnc symbol is only 15602
MF_genes %>% filter(hgnc_symbol != "") %>% filter(hgnc_symbol=="MSLN")

save(MF_genes_ENSEMBL_HGNC,file="/fcrbiouatappn01/export/home/mxu4/reference/MF_genes_ENSEMBL_HGNC_Biomart.rdata")
listDatasets()
```